diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
index 70a32e62b..4dd2df787 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
@@ -67,26 +67,6 @@ function acq_sku_update_8001() {
  * Implements hook_install().
  */
 function acq_sku_install() {
-  Node::create([
-    'type' => 'page',
-    'title' => 'Store',
-    'langcode' => 'en',
-    'uid' => '1',
-    'status' => 1,
-    'path' => '/store',
-    'body' => [
-      'value' => '<p>Welcome to our store</p>',
-      'format' => 'rich_text',
-    ],
-  ])->save();
-
-  MenuLinkContent::create([
-    'title' => 'Store',
-    'link' => ['uri' => 'internal:/store'],
-    'menu_name' => 'main',
-    'weight' => 0,
-  ])->save();
-
   foreach (_acq_sku_indexes() as $index => $data) {
     if (!\Drupal::database()->schema()->indexExists($data['table'], $index)) {
       \Drupal::database()
