diff --git a/modules/acq_cart/js/acq_cart_custom.js b/modules/acq_cart/js/acq_cart_custom.js
index 5c6ca3235..59c5d8283 100644
--- a/modules/acq_cart/js/acq_cart_custom.js
+++ b/modules/acq_cart/js/acq_cart_custom.js
@@ -28,7 +28,9 @@
     acq_cart_mini_cart_xhr = $.ajax({
       url: Drupal.url('mini-cart'),
       success: function (result) {
-        $('#mini-cart-wrapper').html(result);
+        if ($('#cart_notification').length === 0) {
+          $('#mini-cart-wrapper').html(result);
+        }
       }
     });
   };
diff --git a/modules/acq_cart/templates/block--acqminicart.html.twig b/modules/acq_cart/templates/block--acqminicart.html.twig
index 5216a9e2b..53e23726f 100644
--- a/modules/acq_cart/templates/block--acqminicart.html.twig
+++ b/modules/acq_cart/templates/block--acqminicart.html.twig
@@ -1,4 +1,5 @@
 <div class="acq-mini-cart">
   {{ total }}
   <a class="cart-link" href="{{ url('acq_cart.cart') }}"><span class="quantity">{{ quantity }}</span></a>
+  <div id="cart_notification"></div>
 </div>

