diff --git a/modules/acq_cart/src/CartSessionStorage.php b/modules/acq_cart/src/CartSessionStorage.php
index 58d7a14f..d9b4771b 100644
--- a/modules/acq_cart/src/CartSessionStorage.php
+++ b/modules/acq_cart/src/CartSessionStorage.php
@@ -162,6 +162,9 @@ public function pushCart() {
   public function createCart() {
     $customer_id = NULL;
 
+    // @TODO: It seems this customer_id is never used by Magento.
+    // We may need to edit Magento code to associate the cart if customer_id is
+    // given or use the associate endpoint.
     if (!\Drupal::currentUser()->isAnonymous()) {
       $customer_id = \Drupal::currentUser()->getAccount()->acq_customer_id;
     }
@@ -173,4 +176,25 @@ public function createCart() {
     return $cart;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function associateCart($customer_id) {
+    // We first update the session cart.
+    $cart = $this->session->get(self::STORAGE_KEY);
+    if (!$cart) {
+      return;
+    }
+
+    $data = [
+      'cart_id' => $cart->id(),
+      'customer_id' => $customer_id,
+    ];
+    $cart->convertToCustomerCart($data);
+    $this->session->set(self::STORAGE_KEY, $cart);
+
+    // Then we notify the commerce backend about the association.
+    \Drupal::service('acq_commerce.api')->associateCart($cart->id(), $cart->customerId());
+  }
+
 }
diff --git a/modules/acq_cart/src/CartStorageInterface.php b/modules/acq_cart/src/CartStorageInterface.php
index a625adf2..b1c9e30f 100644
--- a/modules/acq_cart/src/CartStorageInterface.php
+++ b/modules/acq_cart/src/CartStorageInterface.php
@@ -43,4 +43,9 @@ public function updateCart();
    */
   public function createCart();
 
+  /**
+   * Associate the current cart in storage with a given customer.
+   */
+  public function associateCart($customer_id);
+
 }
diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 9fe29f70..dd2fa110 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -166,6 +166,41 @@ public function updateCart($cart_id, $cart) {
   }
 
   /**
+   * Associate a cart with a customer.
+   *
+   * @param int $cart_id
+   *   ID of cart to associate.
+   * @param $customer_id
+   *   ID of customer to associate with.
+   *
+   * @return bool
+   *   A status of coupon being applied.
+   * @throws \Exception
+   */
+  public function associateCart($cart_id, $customer_id) {
+    $endpoint = "cart/$cart_id/associate";
+
+    $doReq = function ($client, $opt) use ($endpoint, $customer_id, $cart_id) {
+      $opt['json'] = [
+        'customer_id' => $customer_id,
+        'cart_id' => $cart_id
+      ];
+      return ($client->post($endpoint, $opt));
+    };
+
+    $status = FALSE;
+
+    try {
+      $status = (bool) $this->tryAgentRequest($doReq, 'associateCart');
+    }
+    catch (ConductorException $e) {
+      throw new \Exception($e->getMessage(), $e->getCode());
+    }
+
+    return $status;
+  }
+
+  /**
    * Finalizes a cart's order.
    *
    * @param int $cart_id
