diff --git a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index a442662..e3a19cf 100644
--- a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -281,7 +281,9 @@ public function deriveProductTree(SKU $sku) {
 
     foreach ($sku->field_configured_skus as $child_sku) {
       $child_sku = SKU::loadFromSKU($child_sku->getString());
-      $tree['products'][$child_sku->getSKU()] = $child_sku;
+      if (!empty($child_sku)) {
+        $tree['products'][$child_sku->getSKU()] = $child_sku;
+      }
     }
 
     $configurables = unserialize(
