diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
index 7b4f7c2bc..6c5588c9b 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
@@ -189,7 +189,7 @@ public function getSwatchImage($download = TRUE, $reset = FALSE, $default_label
 
         $media_item = $this->processMediaItem($update_sku, $data, $download, $default_label);
 
-        if ($media_item && in_array(self::SWATCH_IMAGE_ROLE, $media_item['roles'])) {
+        if ($media_item && !empty($media_item['roles']) && in_array(self::SWATCH_IMAGE_ROLE, $media_item['roles'])) {
           $this->swatchData = $media_item;
           break;
         }
