diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
index 635bcaa43..380cc6a63 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
@@ -11,6 +11,21 @@
 use Drupal\menu_link_content\Entity\MenuLinkContent;
 use Symfony\Component\Yaml\Yaml;
 
+/**
+ * Implements hook_update_N().
+ *
+ * Adding index on 'acq_sku__field_configured_skus' table for performance.
+ */
+function acq_sku_update_8007() {
+  $index = 'acq_sku__field_configured_skus';
+  $data = _acq_sku_indexes()[$index];
+  if (!\Drupal::database()->schema()->indexExists($data['table'], $index)) {
+    \Drupal::database()
+      ->schema()
+      ->addIndex($data['table'], $index, $data['fields'], $data['spec']);
+  }
+}
+
 /**
  * Implements hook_update_N().
  *
@@ -257,5 +272,22 @@ function _acq_sku_indexes() {
     ],
   ];
 
+  // Add index for 'field_skus on acq_sku__field_configured_skus' table.
+  $indexes['acq_sku__field_configured_skus'] = [
+    'table' => 'acq_sku__field_configured_skus',
+    'fields' => [
+      ['field_configured_skus_value', 64],
+    ],
+    'spec' => [
+      'fields' => [
+        'field_configured_skus_value' => [
+          'type' => 'varchar',
+          'length' => 255,
+          'default' => '',
+        ],
+      ],
+    ],
+  ];
+
   return $indexes;
 }
