diff --git a/src/Simplesitemap.php b/src/Simplesitemap.php
index de55a53..5f5cb3c 100644
--- a/src/Simplesitemap.php
+++ b/src/Simplesitemap.php
@@ -3,6 +3,7 @@
 namespace Drupal\simple_sitemap;
 
 use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\simple_sitemap\Queue\QueueWorker;
 use Drupal\Core\Path\PathValidator;
