diff --git a/src/Plugin/simple_sitemap/SitemapGenerator/DefaultSitemapGenerator.php b/src/Plugin/simple_sitemap/SitemapGenerator/DefaultSitemapGenerator.php
index e62fda9..c9b31d9 100755
--- a/src/Plugin/simple_sitemap/SitemapGenerator/DefaultSitemapGenerator.php
+++ b/src/Plugin/simple_sitemap/SitemapGenerator/DefaultSitemapGenerator.php
@@ -136,8 +136,8 @@ class DefaultSitemapGenerator extends SitemapGeneratorBase {
         foreach ($link['alternate_urls'] as $language_id => $alternate_url) {
           $this->writer->startElement('xhtml:link');
           $this->writer->writeAttribute('rel', 'alternate');
-          $this->writer->writeAttribute('hreflang', $language_id);
           $this->writer->writeAttribute('href', $alternate_url);
+          $this->writer->writeAttribute('hreflang', $language_id);
           $this->writer->endElement();
         }
       }
diff --git a/src/Simplesitemap.php b/src/Simplesitemap.php
index 5daa00b..dd45c84 100644
--- a/src/Simplesitemap.php
+++ b/src/Simplesitemap.php
@@ -644,6 +644,10 @@ class Simplesitemap {
 
     $entity = $this->entityTypeManager->getStorage($entity_type_id)->load($id);
 
+    if (!$entity instanceof EntityInterface) {
+      return $this;
+    }
+
     $all_bundle_settings = $this->getBundleSettings(
       $entity_type_id, $this->entityHelper->getEntityInstanceBundleName($entity), TRUE, TRUE
     );
diff --git a/src/Simplesitemap.php b/src/Simplesitemap.php
index de55a53..5f5cb3c 100644
--- a/src/Simplesitemap.php
+++ b/src/Simplesitemap.php
@@ -3,6 +3,7 @@
 namespace Drupal\simple_sitemap;
 
 use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\simple_sitemap\Queue\QueueWorker;
 use Drupal\Core\Path\PathValidator;
