diff --git a/clientside_validation_jquery/clientside_validation_jquery.module b/clientside_validation_jquery/clientside_validation_jquery.module
index c4b5a61..adb72f1 100644
--- a/clientside_validation_jquery/clientside_validation_jquery.module
+++ b/clientside_validation_jquery/clientside_validation_jquery.module
@@ -46,6 +46,11 @@ function clientside_validation_jquery_clientside_validation_validator_info_alter
  */
 function clientside_validation_jquery_library_info_alter(&$libraries, $extension) {
   if ($extension == 'clientside_validation_jquery' && isset($libraries['jquery.validate'])) {
+    // Add support for inline_form_errors module.
+    if (\Drupal::moduleHandler()->moduleExists('inline_form_errors')) {
+      $libraries['jquery.validate']['js']['js/cv.jquery.ife.js'] = drupal_get_path('module', 'clientside_validation_jquery') . '/js/cv.jquery.ife.js';
+    }
+
     // Based on readme instructions for composer support, check in libraries.
     if (file_exists('libraries/jqueryvalidate/dist/jquery.validate.js')) {
       return;
diff --git a/clientside_validation_jquery/js/cv.jquery.ife.js b/clientside_validation_jquery/js/cv.jquery.ife.js
new file mode 100644
index 0000000..f2fbc3a
--- /dev/null
+++ b/clientside_validation_jquery/js/cv.jquery.ife.js
@@ -0,0 +1,28 @@
+/**
+ * @file
+ * Attaches behaviors for the Clientside Validation jQuery module.
+ */
+
+(function ($, Drupal) {
+  'use strict';
+
+  /**
+   * Attaches jQuery validate behavior to forms.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *  Attaches the outline behavior to the right context.
+   */
+  Drupal.behaviors.cvJqueryValidateIfe = {
+    attach: function (context) {
+      $(context).find('.form-item input, .form-item select, .form-item textarea').each(function () {
+        $(this).on('focusin', function () {
+          var wrapper = $(this).closest('.form-item');
+          wrapper.find('.form-item--error-message').remove();
+          wrapper.removeClass('form-item--error');
+        });
+      });
+    }
+  };
+})(jQuery, Drupal);
