diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.libraries.yml b/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.libraries.yml
deleted file mode 100644
index 9f6b11e8..00000000
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.libraries.yml
+++ /dev/null
@@ -1,7 +0,0 @@
-acq-cart-custom-js:
-  version: 1.x
-  js:
-    js/acq_cart_custom.js: {}
-  dependencies:
-    - core/jquery
-    - core/drupal
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.routing.yml b/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.routing.yml
index 756cb264..46b2da9c 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.routing.yml
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.routing.yml
@@ -5,11 +5,3 @@ acq_cart.cart:
     _title: 'Cart'
   requirements:
     _permission: 'access cart page'
-
-acq_cart.getMiniCartRender:
-  path: '/mini-cart'
-  defaults:
-    _controller: '\Drupal\acq_cart\Controller\GetMiniCart::content'
-    _title: 'Cart'
-  requirements:
-    _permission: 'access content'
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.services.yml b/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.services.yml
index 085d6716..2c1290ca 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.services.yml
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/acq_cart.services.yml
@@ -2,3 +2,6 @@ services:
   acq_cart.cart_storage:
     class: Drupal\acq_cart\CartSessionStorage
     arguments: ['@session', '@acq_commerce.api', '@logger.factory']
+  acq_cart.mini_cart:
+    class: Drupal\acq_cart\MiniCartManager
+    arguments: ['@acq_cart.cart_storage']
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/js/acq_cart_custom.js b/docroot/modules/commerce/acq_commerce/modules/acq_cart/js/acq_cart_custom.js
deleted file mode 100644
index 5c6ca323..00000000
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/js/acq_cart_custom.js
+++ /dev/null
@@ -1,36 +0,0 @@
-(function ($, Drupal) {
-  'use strict';
-
-  var acq_cart_mini_cart_xhr;
-  var acq_cart_mini_cart_timeout;
-
-  Drupal.behaviors.acq_cart_js = {
-    attach: function (context, settings) {
-      // Do the processing only if we have mini cart available in DOM.
-      if ($('body').find('#mini-cart-wrapper').length) {
-        // Clear time out, behaviors can be called multiple times.
-        if (acq_cart_mini_cart_timeout) {
-          clearTimeout(acq_cart_mini_cart_timeout);
-        }
-
-        // Let other JS operations finish.
-        acq_cart_mini_cart_timeout = setTimeout(Drupal.updateMiniCart, 50);
-      }
-    }
-  };
-
-  Drupal.updateMiniCart = function () {
-    // Abort the AJAX operation if already started before starting new one.
-    if (acq_cart_mini_cart_xhr && acq_cart_mini_cart_xhr.readyState !== 4) {
-      acq_cart_mini_cart_xhr.abort();
-    }
-
-    acq_cart_mini_cart_xhr = $.ajax({
-      url: Drupal.url('mini-cart'),
-      success: function (result) {
-        $('#mini-cart-wrapper').html(result);
-      }
-    });
-  };
-
-})(jQuery, Drupal);
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Controller/GetMiniCart.php b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Controller/GetMiniCart.php
deleted file mode 100644
index d14b5bb6..00000000
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Controller/GetMiniCart.php
+++ /dev/null
@@ -1,80 +0,0 @@
-<?php
-
-namespace Drupal\acq_cart\Controller;
-
-use Drupal\Core\Controller\ControllerBase;
-use Drupal\acq_cart\CartStorageInterface;
-use Symfony\Component\DependencyInjection\ContainerInterface;
-use Symfony\Component\HttpFoundation\JsonResponse;
-use Symfony\Component\HttpFoundation\Response;
-
-/**
- * Class GetMiniCart.
- *
- * @package Drupal\acq_cart\Controller
- */
-class GetMiniCart extends ControllerBase {
-  /**
-   * Drupal\acq_cart\CartStorageInterface definition.
-   *
-   * @var \Drupal\acq_cart\CartStorageInterface
-   */
-  protected $cartStorage;
-
-  /**
-   * Constructor.
-   *
-   * @param \Drupal\acq_cart\CartStorageInterface $cart_storage
-   *   The cart session.
-   */
-  public function __construct(CartStorageInterface $cart_storage) {
-    $this->cartStorage = $cart_storage;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public static function create(ContainerInterface $container) {
-    return new static(
-      $container->get('acq_cart.cart_storage')
-    );
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function content() {
-    $cart = $this->cartStorage->getCart(FALSE);
-
-    // Return empty cart in case no cart available in current session.
-    $output = [
-      '#theme' => 'acq_cart_mini_cart',
-    ];
-
-    if (!empty($cart)) {
-      $totals = $cart->totals();
-
-      // The grand total including discounts and taxes.
-      $grand_total = $totals['grand'] < 0 || $totals['grand'] == NULL ? 0 : $totals['grand'];
-
-      // Deduct shipping.
-      if (isset($totals['shipping']) && $grand_total) {
-        $grand_total -= $totals['shipping'];
-      }
-
-      $total = [
-        '#theme' => 'acq_commerce_price',
-        '#price' => $grand_total,
-      ];
-
-      // Use the template to render the HTML.
-      $output['#quantity'] = $cart->getCartItemsCount();
-      $output['#total'] = render($total);
-    }
-
-    $response = new Response();
-    $response->setContent(render($output));
-    return $response;
-  }
-
-}
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/MiniCartManager.php b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/MiniCartManager.php
new file mode 100644
index 00000000..7a53a0e6
--- /dev/null
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/MiniCartManager.php
@@ -0,0 +1,59 @@
+<?php
+
+namespace Drupal\acq_cart;
+
+/**
+ * Class MiniCartManager.
+ *
+ * @package Drupal\acq_cart
+ */
+class MiniCartManager {
+
+  /**
+   * MiniCartManager constructor.
+   *
+   * @param \Drupal\acq_cart\CartStorageInterface $cartStorage
+   *   Cart storage service.
+   */
+  public function __construct(CartStorageInterface $cartStorage) {
+    $this->cartStorage = $cartStorage;
+  }
+
+  /**
+   * Helper function to get Mini cart.
+   */
+  public function getMiniCart() {
+    $cart = $this->cartStorage->getCart(FALSE);
+
+    // Return empty cart in case no cart available in current session.
+    $output = [
+      '#theme' => 'acq_cart_mini_cart',
+    ];
+
+    if (!empty($cart)) {
+      $totals = $cart->totals();
+
+      // The grand total including discounts and taxes.
+      $grand_total = $totals['grand'] < 0 || $totals['grand'] == NULL ? 0 : $totals['grand'];
+
+      // Deduct shipping.
+      if (isset($totals['shipping']) && $grand_total) {
+        $grand_total -= $totals['shipping'];
+      }
+
+      $total = [
+        '#theme' => 'acq_commerce_price',
+        '#price' => $grand_total,
+      ];
+
+      // Use the template to render the HTML.
+      $output['#quantity'] = $cart->getCartItemsCount();
+      $output['#total'] = render($total);
+    }
+
+    $output['#prefix'] = '<div id="mini-cart-wrapper">';
+    $output['#suffix'] = '</div><div id="cart_notification"></div>';
+    return $output;
+  }
+
+}
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Plugin/Block/CartMiniBlock.php b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Plugin/Block/CartMiniBlock.php
index 8c66aed1..87161205 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Plugin/Block/CartMiniBlock.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/Plugin/Block/CartMiniBlock.php
@@ -32,13 +32,11 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function build() {
     // Something to show till we get the AJAX response back.
-    $loading_message = $this->t('Loading ...');
     $output = [
-      '#markup' => '<div id="mini-cart-wrapper">' . $loading_message . '</div><div id="cart_notification"></div>',
-      '#attached' => [
-        'library' => 'acq_cart/acq-cart-custom-js',
-      ],
+      '#lazy_builder' => ['acq_cart.mini_cart:getMiniCart', array()],
+      '#create_placeholder' => TRUE
     ];
+
     return $output;
   }
 
