--- core/modules/views/js/ajax_view.js	(revision )
+++ core/modules/views/js/ajax_view.js	(revision )
@@ -131,7 +131,8 @@
     $('input[type=submit], input[type=image]', this.$exposed_form).not('[data-drupal-selector=edit-reset]').each(function (index) {
       var self_settings = $.extend({}, that.element_settings, {
         base: $(this).attr('id'),
-        element: this
+        element: this,
+        type: 'GET'
       });
       that.exposedFormAjax[index] = Drupal.ajax(self_settings);
     });
