diff --git a/.htaccess b/.htaccess
index 1dad3c1..dc494a7 100644
--- a/.htaccess
+++ b/.htaccess
@@ -184,6 +184,8 @@ AddEncoding gzip svgz
   # Copy and adapt this rule to directly execute PHP files in contributed or
   # custom modules or to run another PHP application in the same directory.
   RewriteCond %{REQUEST_URI} !/core/modules/statistics/statistics\.php$
+  # Allow access to simplesaml paths
+  RewriteCond %{REQUEST_URI} !^/simplesaml
   # ACSF requirement: allow access to apc_rebuild.php.
   RewriteCond %{REQUEST_URI} !/sites/g/apc_rebuild.php$
   # Deny access to any other PHP files that do not match the rules above.
