diff --git a/.htaccess b/.htaccess
index 83cb71a04..ffcc2bbfd 100644
--- a/.htaccess
+++ b/.htaccess
@@ -138,6 +138,8 @@ AddEncoding gzip svgz
   RewriteCond %{REQUEST_URI} !=/favicon.ico
   RewriteRule ^ index.php [L]

+  # Allow access to simplesaml paths
+  RewriteCond %{REQUEST_URI} !^/simplesaml
   # For security reasons, deny access to other PHP files on public sites.
   # Note: The following URI conditions are not anchored at the start (^),
   # because Drupal may be located in a subdirectory. To further improve
