diff --git a/.htaccess b/.htaccess
index 91bb98a..7b93e45 100644
--- a/.htaccess
+++ b/.htaccess
@@ -69,6 +69,10 @@ AddEncoding gzip svgz
 <IfModule mod_rewrite.c>
   RewriteEngine on
 
+  # Remove trailing slash from all URLs (except home, which is practically not possible).
+  RewriteCond %{REQUEST_FILENAME} !-d
+  RewriteRule ^(.*)/$ /$1 [L,R=301]
+
   # Set "protossl" to "s" if we were accessed via https://.  This is used later
   # if you enable "www." stripping or enforcement, in order to ensure that
   # you don't bounce between http and https.
