diff --git a/.htaccess b/.htaccess
index 83cb71a04..4885ce5c6 100644
--- a/.htaccess
+++ b/.htaccess
@@ -75,6 +75,8 @@ AddEncoding gzip svgz
   RewriteRule ^ - [E=protossl]
   RewriteCond %{HTTPS} on
   RewriteRule ^ - [E=protossl:s]
+  RewriteCond %{HTTP:X-Forwarded-Proto} https
+  RewriteRule ^ - [E=protossl:s]

   # Make sure Authorization HTTP header is available to PHP
   # even when running as CGI or FastCGI.
