diff --git a/.htaccess b/.htaccess
index 83cb71a04..9fc946b67 100644
--- a/.htaccess
+++ b/.htaccess
@@ -111,6 +111,23 @@ AddEncoding gzip svgz
   # uncomment the following:
   # RewriteCond %{HTTP_HOST} ^www\.(.+)$ [NC]
   # RewriteRule ^ http%{ENV:protossl}://%1%{REQUEST_URI} [L,R=301]
+  RewriteCond %{HTTP_HOST} .
+  RewriteCond %{HTTP_HOST} !^www\. [NC]
+  # Don't redirect ACSF domains.
+  RewriteCond %{HTTP_HOST} !\.acsitefactory\.com$ [NC]
+  # Don't redirect preprod domains.
+  RewriteCond %{HTTP_HOST} !\.factory\.alshaya\.com$ [NC]
+  # Don't redirect local dev sites.
+  RewriteCond %{HTTP_HOST} !^local\. [NC]
+  RewriteCond %{HTTP_HOST} !\.site$ [NC]
+  RewriteCond %{HTTP_HOST} !\.cloud$ [NC]
+  # Don't redirect on all *.hm.com domains.
+  RewriteCond %{HTTP_HOST} !\.hm\.com$ [NC]
+  # Don't redirect on all *.boots.com domains.
+  RewriteCond %{HTTP_HOST} !\.boots\.com$ [NC]
+  # Don't redirect on all *.cosstores.com domains.
+  RewriteCond %{HTTP_HOST} !\.cosstores\.com$ [NC]
+  RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

   # Modify the RewriteBase if you are using Drupal in a subdirectory or in a
   # VirtualDocumentRoot and the rewrite rules are not working properly.
