diff --git a/.htaccess b/.htaccess
index dcd85eb..0ee33cb 100644
--- a/.htaccess
+++ b/.htaccess
@@ -132,6 +132,12 @@ AddEncoding gzip svgz
   # Redirect HTTP to HTTPS for all requests
   RewriteCond %{HTTPS} off
   RewriteCond %{HTTP:X-Forwarded-Proto} !https [novary]
+  # Make sure we are not executing this rule after an internal redirection for
+  # index.php. Note that variable name is prefixed by Apache when recomputing
+  # redirect rules after an internal redirection.
+  RewriteCond %{ENV:REDIRECT_internal_redirect} !true
+  RewriteCond %{REQUEST_URI} !(knet) [NC]
+  RewriteCond %{REQUEST_URI} !(/forwarded) [NC]
   RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

   # Redirect common PHP files to their new locations.
@@ -148,7 +154,7 @@ AddEncoding gzip svgz
   RewriteCond %{REQUEST_FILENAME} !-f
   RewriteCond %{REQUEST_FILENAME} !-d
   RewriteCond %{REQUEST_URI} !=/favicon.ico
-  RewriteRule ^ index.php [L]
+  RewriteRule ^ index.php [L,E=internal_redirect:true]

   # For security reasons, deny access to other PHP files on public sites.
   # Note: The following URI conditions are not anchored at the start (^),
