diff --git a/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php b/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php
--- a/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php	(date 1614346426297)
+++ b/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php	(date 1614346426297)
@@ -371,9 +371,12 @@
           return;
         }

-        foreach ($ids as $id) {
-          $this->getAlgoliaIndex()->deleteObject($id);
-        }
+        $response = $this->getAlgoliaIndex()->deleteObjects($ids);
+        $this->getLogger()->notice('Deletion requested for IDs: @ids on Algolia for Language: @language, Response: @response.', [
+          '@response' => json_encode($response),
+          '@language' => $language->getId(),
+          '@ids' => implode(',', $ids),
+        ]);
       }
     }
     else {
@@ -389,9 +392,11 @@
         return;
       }

-      foreach ($ids as $id) {
-        $this->getAlgoliaIndex()->deleteObject($id);
-      }
+      $response = $this->getAlgoliaIndex()->deleteObjects($ids);
+      $this->getLogger()->notice('Deletion requested for IDs: @ids on Algolia. Response: @response.', [
+        '@response' => json_encode($response),
+        '@ids' => implode(',', $ids),
+      ]);
     }
   }

@@ -406,7 +411,11 @@
           $this->connect($index, '', $language->getId());

           // Clearing the full index.
-          $this->getAlgoliaIndex()->clearIndex();
+          $response = $this->getAlgoliaIndex()->clearIndex();
+          $this->getLogger()->notice('Deletion requested for full index on Algolia for Language: @language, Response: @response.', [
+            '@response' => json_encode($response),
+            '@language' => $language->getId(),
+          ]);
         }
       }
       else {
@@ -414,7 +423,10 @@
         $this->connect($index);

         // Clearing the full index.
-        $this->getAlgoliaIndex()->clearIndex();
+        $response = $this->getAlgoliaIndex()->clearIndex();
+        $this->getLogger()->notice('Deletion requested for full index on Algolia. Response: @response.', [
+          '@response' => json_encode($response),
+        ]);
       }
     }
   }
