diff --git a/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php b/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php
index 2118e39..0d9c6a8 100644
--- a/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php
+++ b/src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php
@@ -9,6 +9,7 @@ use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
 use Drupal\Core\Language\LanguageManagerInterface;
 use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\Core\Site\Settings;
 use Drupal\search_api\Backend\BackendPluginBase;
 use Drupal\search_api\Item\ItemInterface;
 use Drupal\search_api\IndexInterface;
@@ -216,6 +217,16 @@ class SearchApiAlgoliaBackend extends BackendPluginBase implements PluginFormInt
       }
 
       foreach ($itemsToIndex as $language => $items) {
+        // Allow adding objects to logs for investigation.
+        if (Settings::get('search_api_algolia_debug', 0)) {
+          foreach ($items as $item) {
+            $this->getLogger()->notice('Data pushed to Algolia for Language @language : @data', [
+              '@data' => json_encode($item),
+              '@language' => $language,
+            ]);
+          }
+        }
+
         try {
           $this->connect($index, '', $language);
           $this->getAlgoliaIndex()->saveObjects($items);
