diff --git a/core/lib/Drupal/Core/DrupalKernel.php b/core/lib/Drupal/Core/DrupalKernel.php
index 69972b1cce..0d5d282643 100644
--- a/core/lib/Drupal/Core/DrupalKernel.php
+++ b/core/lib/Drupal/Core/DrupalKernel.php
@@ -701,6 +701,15 @@ public function handle(Request $request, $type = self::MASTER_REQUEST, $catch =
       // installed yet (i.e., if no $databases array has been defined in the
       // settings.php file) and we are not already installing.
       if (!Database::getConnectionInfo() && !InstallerKernel::installationAttempted() && PHP_SAPI !== 'cli') {
+        if (Settings::get('alshaya_installer_redirection_log', FALSE)) {
+          $this->container->get('logger.factory')->get('DrupalKernel')->error(
+            'The user is redirected to "core/install.php". Error: @message Requested from @trace',
+            [
+              '@message' => $e->getMessage(),
+              '@trace' => json_encode(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 10))
+            ]
+          );
+        }
         $response = new RedirectResponse($request->getBasePath() . '/core/install.php', 302, ['Cache-Control' => 'no-cache']);
       }
       else {
@@ -741,6 +750,15 @@ public function handle(Request $request, $type = self::MASTER_REQUEST, $catch =
    */
   protected function handleException(\Exception $e, $request, $type) {
     if ($this->shouldRedirectToInstaller($e, $this->container ? $this->container->get('database') : NULL)) {
+      if (Settings::get('alshaya_installer_redirection_log', FALSE)) {
+        $this->container->get('logger.factory')->get('DrupalKernel')->error(
+          'The user is redirected to "core/install.php". Error: @message Requested from @trace',
+          [
+            '@message' => $e->getMessage(),
+            '@trace' => json_encode(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 10))
+          ]
+        );
+      }
       return new RedirectResponse($request->getBasePath() . '/core/install.php', 302, ['Cache-Control' => 'no-cache']);
     }
 
