diff --git a/core/lib/Drupal/Core/Asset/CssCollectionOptimizer.php b/core/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
index 26ed07f79e..5766309c33 100644
--- a/core/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
+++ b/core/lib/Drupal/Core/Asset/CssCollectionOptimizer.php
@@ -134,6 +134,12 @@ class CssCollectionOptimizer implements AssetCollectionOptimizerInterface {
               $data = implode('', $matches[0]) . $data;
               // Dump the optimized CSS for this group into an aggregate file.
               $uri = $this->dumper->dump($data, 'css');
+              // CORE-34870: Add Log for Debugging.
+                \Drupal::logger('CssCollectionOptimizer')->notice('Optimized CSS file is dumped on @uri, with key = @key , CSS Group = @group', [
+                  '@uri' => $uri,
+                  '@key' => $key,
+                  '@group' => json_encode($css_group),
+              ]);
               // Set the URI for this group's aggregate file.
               $css_assets[$order]['data'] = $uri;
               // Persist the URI for this aggregate file.
diff --git a/core/lib/Drupal/Core/Asset/JsCollectionOptimizer.php b/core/lib/Drupal/Core/Asset/JsCollectionOptimizer.php
index dc66652b6c..01dc71407c 100644
--- a/core/lib/Drupal/Core/Asset/JsCollectionOptimizer.php
+++ b/core/lib/Drupal/Core/Asset/JsCollectionOptimizer.php
@@ -137,6 +137,12 @@ class JsCollectionOptimizer implements AssetCollectionOptimizerInterface {
               $data = $this->optimizer->clean($data);
               // Dump the optimized JS for this group into an aggregate file.
               $uri = $this->dumper->dump($data, 'js');
+              // CORE-34870: Add Log for Debugging.
+              \Drupal::logger('JsCollectionOptimizer')->notice('Optimized JS file is dumped on @uri, with key = @key , JS Group = @group', [
+                  '@uri' => $uri,
+                  '@key' => $key,
+                  '@group' => json_encode($js_group),
+              ]);
               // Set the URI for this group's aggregate file.
               $js_assets[$order]['data'] = $uri;
               // Persist the URI for this aggregate file.
