diff --git a/search_api.api.php b/search_api.api.php
index 9e36b45c..3cb88a6a 100644
--- a/search_api.api.php
+++ b/search_api.api.php
@@ -432,6 +432,34 @@ function hook_search_api_index_reindex(\Drupal\search_api\IndexInterface $index,
     ->execute();
 }
 
+/**
+ * Alter the item_ids for entity delete.
+ *
+ * @param array $item_ids
+ *   Array of query parameters.
+ * @param object $index
+ *   Object of index.
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   EntityInterface of entity.
+ */
+function hook_alshaya_search_api_entity_delete_alter(array &$item_ids, $index, EntityInterface $entity) {
+
+}
+
+/**
+ * Alter the item_ids for entity delete.
+ *
+ * @param array $item_ids
+ *   Array of query parameters.
+ * @param object $index
+ *   Object of index.
+ * @param string $id
+ *   String of query parameters.
+ */
+function hook_alshaya_search_api_track_items_alter(array &$item_ids, $index, $id) {
+
+}
+
 /**
  * @} End of "addtogroup hooks".
  */
diff --git a/src/Entity/Index.php b/src/Entity/Index.php
index 090131de..9f0ed373 100644
--- a/src/Entity/Index.php
+++ b/src/Entity/Index.php
@@ -1105,6 +1105,7 @@ class Index extends ConfigEntityBase implements IndexInterface {
     foreach ($ids as $id) {
       $item_ids[] = Utility::createCombinedId($datasource_id, $id);
     }
+    \Drupal::moduleHandler()->alter('alshaya_search_api_track_items', $item_ids, $this, $id);
     if ($this->hasValidTracker()) {
       $this->getTrackerInstance()->trackItemsDeleted($item_ids);
     }
diff --git a/src/Plugin/search_api/datasource/ContentEntityTrackingManager.php b/src/Plugin/search_api/datasource/ContentEntityTrackingManager.php
index 551d2116..90069aa1 100644
--- a/src/Plugin/search_api/datasource/ContentEntityTrackingManager.php
+++ b/src/Plugin/search_api/datasource/ContentEntityTrackingManager.php
@@ -202,6 +202,7 @@ class ContentEntityTrackingManager {
     }
     $datasource_id = 'entity:' . $entity->getEntityTypeId();
     foreach ($indexes as $index) {
+      \Drupal::moduleHandler()->alter('alshaya_search_api_entity_delete', $item_ids, $index, $entity);
       $index->trackItemsDeleted($datasource_id, $item_ids);
     }
   }
