diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AlshayaPromotionQueueBase.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionQueueBase.php
similarity index 96%
rename from docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AlshayaPromotionQueueBase.php
rename to docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionQueueBase.php
index d06ef4e0..206000bd 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AlshayaPromotionQueueBase.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionQueueBase.php
@@ -15,7 +15,7 @@
  *
  * @package Drupal\acq_promotion
  */
-abstract class AlshayaPromotionQueueBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+abstract class AcqPromotionQueueBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
   use IngestRequestTrait;
 
   /**
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
index 588b4da1..1b4780ff 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
@@ -2,9 +2,8 @@
 
 namespace Drupal\acq_promotion\Plugin\QueueWorker;
 
-use Drupal\acq_promotion\AlshayaPromotionQueueBase;
+use Drupal\acq_promotion\AcqPromotionQueueBase;
 use Drupal\acq_sku\Entity\SKU;
-use Drupal\node\Entity\Node;
 
 /**
  * Processes Skus to attach Promotions.
@@ -14,7 +13,7 @@
  *   title = @Translation("Acq Commerce Promotion attach queue"),
  * )
  */
-class AcqPromotionAttachQueue extends AlshayaPromotionQueueBase {
+class AcqPromotionAttachQueue extends AcqPromotionQueueBase {
 
   /**
    * Works on a single queue item.
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
index 56aeaa3f..c2fdb01b 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
@@ -3,7 +3,7 @@
 namespace Drupal\acq_promotion\Plugin\QueueWorker;
 
 use Drupal\acq_commerce\Conductor\ConductorException;
-use Drupal\acq_promotion\AlshayaPromotionQueueBase;
+use Drupal\acq_promotion\AcqPromotionQueueBase;
 use Drupal\acq_sku\Entity\SKU;
 
 /**
@@ -14,7 +14,7 @@
  *   title = @Translation("Acq Commerce Promotion detach queue"),
  * )
  */
-class AcqPromotionDetachQueue extends AlshayaPromotionQueueBase {
+class AcqPromotionDetachQueue extends AcqPromotionQueueBase {
 
   /**
    * Works on a single queue item.
