diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
index e414de95a..7f2326947 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
@@ -113,7 +113,7 @@ public function getMedia($reset = FALSE) {
       }
 
       foreach ($media_data as &$data) {
-        if ($data['media_type'] == 'image') {
+        if (isset($data['media_type']) && $data['media_type'] == 'image') {
           if (empty($data['fid'])) {
             try {
               // Prepare the File object when we access it the first time.
@@ -203,7 +203,7 @@ public function getThumbnail() {
 
     // We loop through all the media items and return the first image.
     foreach ($media as $media_item) {
-      if ($media_item['media_type'] == 'image') {
+      if (isset($media_item['media_type']) && $media_item['media_type'] == 'image') {
         return $media_item;
       }
     }
