diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
index 66defbcca..af0fc9253 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
@@ -68,12 +68,14 @@ function acq_sku_add_base_fields(array $fields) {
   foreach ($fields as $machine_name => $field) {
     // Confirm we have minimum required data available to create the fields.
     if (empty($field['label'])) {
-      throw new \Exception('Field Label cannot be empty.');
+      \Drupal::logger('acq_sku')->error('Unable to create field @machine_name because field label is empty.', ['@machine_name' => $machine_name]);
+      continue;
     }
 
     // Check if the field is already available in config.
     if (isset($additionalFields[$machine_name])) {
-      throw new \Exception('Field machine name used already.');
+      \Drupal::logger('acq_sku')->notice('Field @machine_name already exists.', ['@machine_name' => $machine_name]);
+      continue;
     }
 
     // Add the field to storedFields so we store them to config.
