diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
index 701dc370f1ec..c64a8e4b3d69 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
@@ -2,10 +2,10 @@

 namespace Drupal\acq_sku\AcquiaCommerce;

+use Drupal\Component\Render\FormattableMarkup;
 use Drupal\Core\Form\FormInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\acq_sku\Entity\SKU;
-use Drupal\Core\Language\LanguageInterface;
 use Drupal\node\Entity\Node;
 use Drupal\Core\Link;
