diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartSessionStorage.php b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartSessionStorage.php
index 2569ec603..5b04b2e66 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartSessionStorage.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartSessionStorage.php
@@ -224,6 +224,8 @@ public function updateCart($create_new = TRUE) {
         $cart = new Cart($cartObject);
       }
 
+      Cache::invalidateTags(['cart:' . $cart_id]);
+
       $this->addCart($cart);
 
       return $cart;
