diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
index 0cc186f12..447f9496f 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
@@ -169,11 +169,13 @@ protected function unpublishPromotions(array $validIDs = []) {
 
       // Detach promotion from all skus.
       $attached_promotion_skus = $this->getSkusForPromotion($node);\
+
       if (!empty($attached_promotion_skus)) {
         $chunks = array_chunk($attached_promotion_skus, $acq_promotion_attach_batch_size);
         foreach ($chunks as $chunk) {
           $data['skus'] = $chunk;
           $data['promotion'] = $nid;
+          $data['promotion_type'] = $node->get('field_acq_promotion_type')->getString();
           $acq_promotion_detach_queue = $this->queue->get('acq_promotion_detach_queue');
           $acq_promotion_detach_queue->createItem($data);
         }
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
index 4d77d4b58..6c5f59f27 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
@@ -44,6 +44,7 @@ class AcqPromotionDetachQueue extends AcqPromotionQueueBase {
   public function processItem($data) {
     $skus = $data['skus'];
     $promotion_nid = $data['promotion'];
+    $promotion_type = $data['promotion_type'];
     $promotion_detach_item[] = ['target_id' => $promotion_nid];
 
     foreach ($skus as $sku) {
@@ -69,7 +70,9 @@ public function processItem($data) {
 
     $sku_texts = implode(',', $skus);
 
-    if ($sku_texts) {
+    // The skus detached from a catalog promotion are not part of those coming back during promotions sync.
+    // So their final_price value does not get updated at that time. We thus need to resync these specific skus.
+    if ($promotion_type === 'category' && $sku_texts) {
       foreach (acq_commerce_get_store_language_mapping() as $langcode => $store_id) {
         $this->ingestApiWrapper->productFullSync($store_id, $langcode, $sku_texts);
       }