diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
index f1f6b2812..a53054555 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.module
@@ -148,10 +148,12 @@ function acq_sku_get_sku_stock(SKU $sku_entity, $reset = FALSE) {

       // If cache multiplier is zero we don't cache the stock.
       if ($cache_multiplier = \Drupal::config('acq_sku.settings')->get('stock_cache_multiplier')) {
-        $time = $stock['quantity'] ? $stock['quantity'] * $cache_multiplier : $cache_multiplier;
+        $default_cache_lifetime = $stock['quantity'] ? $stock['quantity'] * $cache_multiplier : $cache_multiplier;
+        $max_cache_lifetime = \Drupal::config('acq_sku.settings')->get('stock_cache_max_lifetime');

         // Calculate the timestamp when we want the cache to expire.
-        $expire = \Drupal::time()->getRequestTime() + $time;
+        $stock_cache_lifetime = min($default_cache_lifetime, $max_cache_lifetime);
+        $expire = $stock_cache_lifetime + \Drupal::time()->getRequestTime();

         // Set the stock in cache.
         \Drupal::cache('stock')->set($cid, $stock, $expire);
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.routing.yml b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.routing.yml
index 2047911d8..d20c55d16 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.routing.yml
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.routing.yml
@@ -60,3 +60,11 @@ acq_sku.sync:
     _title: 'Synchronize Product / Category Data'
   requirements:
     _permission: 'access commerce administration pages'
+
+acq_sku.settings:
+  path: '/admin/commerce/sku/settings'
+  defaults:
+    _form: '\Drupal\acq_sku\Form\SKUSettingsForm'
+    _title: 'SKU Settings Form'
+  requirements:
+    _permission: 'access commerce administration pages'
\ No newline at end of file
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.settings.yml b/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.settings.yml
index b67a4972e..b1a4acd8f 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.settings.yml
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.settings.yml
@@ -1 +1,2 @@
 stock_cache_multiplier: 600
+stock_cache_max_lifetime: 7200
\ No newline at end of file
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Form/SKUSettingsForm.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Form/SKUSettingsForm.php
index d781bdea0..fa20b4075 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Form/SKUSettingsForm.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Form/SKUSettingsForm.php
@@ -2,7 +2,7 @@

 namespace Drupal\acq_sku\Form;

-use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\ConfigFormBase;
 use Drupal\Core\Form\FormStateInterface;

 /**
@@ -12,7 +12,18 @@
  *
  * @ingroup acq_sku
  */
-class SKUSettingsForm extends FormBase {
+class SKUSettingsForm extends ConfigFormBase {
+
+  /**
+   * Gets the configuration names that will be editable.
+   *
+   * @return array
+   *   An array of configuration object names that are editable if called in
+   *   conjunction with the trait's config() method.
+   */
+  protected function getEditableConfigNames() {
+    return ['acq_sku.settings'];
+  }

   /**
    * Returns a unique string identifying the form.
@@ -33,7 +44,10 @@ public function getFormId() {
    *   An associative array containing the current state of the form.
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    // Empty implementation of the abstract submit class.
+    $sku_settings = $this->config('acq_sku.settings');
+    $sku_settings->set('stock_cache_multiplier', $form_state->getValue('stock_cache_multiplier'));
+    $sku_settings->set('stock_cache_max_lifetime', $form_state->getValue('stock_cache_max_lifetime'));
+    $sku_settings->save();
   }

   /**
@@ -48,7 +62,23 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
    *   Form definition array.
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $form['sku_settings']['#markup'] = 'Settings form for SKU. Manage field settings here.';
+    $form = parent::buildForm($form, $form_state);
+
+    $sku_settings = $this->config('acq_sku.settings');
+    $form['stock_cache_multiplier'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Stock Cache Multiplier'),
+      '#description' => $this->t("Multiplier used while generating the cache expiration for SKU's stock. e.g., cache_expire = stock_quantity * stock_cache_multiplier"),
+      '#default_value' => $sku_settings->get('stock_cache_multiplier'),
+    ];
+
+    $form['stock_cache_max_lifetime'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Stock Cache Max Lifetime'),
+      '#description' => $this->t("Maximum lifetime for the stock cache in seconds."),
+      '#default_value' => $sku_settings->get('stock_cache_max_lifetime'),
+    ];
+
     return $form;
   }
