diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
index 2e2bf8eb..60c8bee1 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_promotion/src/AcqPromotionsManager.php
@@ -107,11 +107,6 @@ protected function syncPromotion(array $promotion) {
       /* @var $node \Drupal\node\Entity\Node */
       $node = $this->nodeStorage->load(reset($nids));
 
-      if (serialize($promotion) == $node->get('field_acq_promotion_data')->getString()) {
-        // Promotion data from API matches what is already stored, not updating.
-        return;
-      }
-
       $this->logger->info('Updating promotion for rule id @rule_id', ['@rule_id' => $promotion['rule_id']]);
     }
 
