diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
index 2d127e232..b01e07fa8 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
@@ -106,4 +106,13 @@ function acq_sku_update_8102() {
   $config_path = drupal_get_path('module', 'acq_sku') . '/config/install/acq_sku.configurable_form_settings.yml';
   $data = Yaml::parse($config_path);
   \Drupal::configFactory()->getEditable('acq_sku.configurable_form_settings')->setData($data)->save(TRUE);
+}
+
+/**
+ * Implements hook_update_N().
+ */
+function acq_sku_update_8103() {
+  $config_path = drupal_get_path('module', 'acq_sku') . '/config/install/acq_sku.configurable_form_settings.yml';
+  $data = Yaml::parse($config_path);
+  \Drupal::configFactory()->getEditable('acq_sku.configurable_form_settings')->setData($data)->save(TRUE);
 }
\ No newline at end of file
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.configurable_form_settings.yml b/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.configurable_form_settings.yml
index 6fa25b4eb..5ff5a5bce 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.configurable_form_settings.yml
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/config/install/acq_sku.configurable_form_settings.yml
@@ -1,3 +1,5 @@
 attribute_weights:
   size: 1
-show_quantity: TRUE
\ No newline at end of file
+show_quantity: TRUE
+sortable_options:
+  - size
\ No newline at end of file
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index bc9c3f3cc..05df554bc 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -59,20 +59,26 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
 
       // Sort the options.
       if (!empty($options)) {
-        $query = \Drupal::database()->select('taxonomy_term_field_data', 'ttfd');
-        $query->fields('ttfd', ['tid', 'weight']);
-        $query->join('taxonomy_term__field_sku_attribute_code', 'ttfsac', 'ttfsac.entity_id = ttfd.tid');
-        $query->join('taxonomy_term__field_sku_option_id', 'ttfsoi', 'ttfsoi.entity_id = ttfd.tid');
-        $query->fields('ttfsoi', ['field_sku_option_id_value']);
-        $query->condition('ttfd.vid', ProductOptionsManager::PRODUCT_OPTIONS_VOCABULARY);
-        $query->condition('ttfsac.field_sku_attribute_code_value', $attribute_code);
-        $query->condition('ttfsoi.field_sku_option_id_value', array_keys($options), 'IN');
-        $query->distinct();
-        $query->orderBy('weight', 'ASC');
-        $tids = $query->execute()->fetchAllAssoc('tid');
-
-        foreach ($tids as $tid => $values) {
-          $sorted_options[$values->field_sku_option_id_value] = $options[$values->field_sku_option_id_value];
+        $sortable_attributes = $configurable_form_settings->get('sortable_options');
+        if (in_array($attribute_code, $sortable_attributes)) {
+          $query = \Drupal::database()->select('taxonomy_term_field_data', 'ttfd');
+          $query->fields('ttfd', ['tid', 'weight']);
+          $query->join('taxonomy_term__field_sku_attribute_code', 'ttfsac', 'ttfsac.entity_id = ttfd.tid');
+          $query->join('taxonomy_term__field_sku_option_id', 'ttfsoi', 'ttfsoi.entity_id = ttfd.tid');
+          $query->fields('ttfsoi', ['field_sku_option_id_value']);
+          $query->condition('ttfd.vid', ProductOptionsManager::PRODUCT_OPTIONS_VOCABULARY);
+          $query->condition('ttfsac.field_sku_attribute_code_value', $attribute_code);
+          $query->condition('ttfsoi.field_sku_option_id_value', array_keys($options), 'IN');
+          $query->distinct();
+          $query->orderBy('weight', 'ASC');
+          $tids = $query->execute()->fetchAllAssoc('tid');
+
+          foreach ($tids as $tid => $values) {
+            $sorted_options[$values->field_sku_option_id_value] = $options[$values->field_sku_option_id_value];
+          }
+        }
+        else {
+          $sorted_options = $options;
         }
 
         $form['ajax']['configurables'][$attribute_code] = [
@@ -107,6 +113,7 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
       '#default_value' => 1,
       '#access' => $configurable_form_settings->get('show_quantity'),
       '#required' => TRUE,
+      '#access' => $configurable_form_settings->get('show_quantity'),
       '#size' => 2,
     ];
 
