diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
index 879836ce7..609790df9 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/acq_sku.install
@@ -31,4 +31,16 @@ function acq_sku_install() {
     'menu_name' => 'main',
     'weight' => 0,
   ])->save();
+
+  // Add index for first 30 character of SKU on acq_sku_field_data table.
+  \Drupal::database()->schema()->addIndex('acq_sku_field_data', 'acq_sku_field_sku', [['sku', 30]], [
+    'fields' => [
+      'sku' => [
+        'type' => 'varchar',
+        'length' => 255,
+        'default' => '',
+      ],
+    ],
+  ]);
 }
+
