diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index 431029d..fec5fb4 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -219,6 +219,9 @@ public function addToCartSubmit(&$form, FormStateInterface $form_state) {
 
       $cart->addItemToCart($tree_pointer->getSKU(), $quantity);
 
+      // Add child SKU to form state to allow other modules to use it.
+      $form_state->setTemporaryValue('child_sku', $tree_pointer->getSKU());
+
       try {
         \Drupal::service('acq_cart.cart_storage')->updateCart();
       }
