diff --git a/modules/acq_sku/src/Entity/SKU.php b/modules/acq_sku/src/Entity/SKU.php
index ba4e445..be05b54 100644
--- a/modules/acq_sku/src/Entity/SKU.php
+++ b/modules/acq_sku/src/Entity/SKU.php
@@ -345,6 +345,18 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDisplayConfigurable('form', TRUE)
       ->setDisplayConfigurable('view', TRUE);
 
+    $fields['special_price'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Special Price'))
+      ->setDescription(t('Special Price of this SKU.'))
+      ->setDisplayConfigurable('form', TRUE)
+      ->setDisplayConfigurable('view', TRUE);
+
+    $fields['final_price'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Final Price'))
+      ->setDescription(t('Final Price of this SKU.'))
+      ->setDisplayConfigurable('form', TRUE)
+      ->setDisplayConfigurable('view', TRUE);
+
     $fields['crosssell'] = BaseFieldDefinition::create('string')
       ->setLabel(t('Cross sell SKU'))
       ->setDescription(t('Reference to all Cross sell SKUs.'))
diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index cb00a5a..cda8a10 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -206,6 +206,9 @@ public function post(array $products = []) {
 
         $sku->name->value = $product['name'];
         $sku->price->value = $product['price'];
+        $sku->special_price->value = $product['special_price'];
+        $sku->final_price->value = $product['final_price'];
+
         $sku->attributes = $this->formatProductAttributes($product['attributes']);
 
         $this->logger->info(
@@ -221,6 +224,8 @@ public function post(array $products = []) {
           'sku' => $product['sku'],
           'name' => html_entity_decode($product['name']),
           'price' => $product['price'],
+          'special_price' => $product['special_price'],
+          'final_price' => $product['final_price'],
           'attributes' => $this->formatProductAttributes($product['attributes']),
         ]);
 
