diff --git a/docroot/modules/commerce/acq_commerce/config/install/acq_commerce.conductor.yml b/docroot/modules/commerce/acq_commerce/config/install/acq_commerce.conductor.yml
index 32374dd..5525422 100644
--- a/docroot/modules/commerce/acq_commerce/config/install/acq_commerce.conductor.yml
+++ b/docroot/modules/commerce/acq_commerce/config/install/acq_commerce.conductor.yml
@@ -4,3 +4,4 @@ timeout: 30
 verify_ssl: true
 product_page_size: 50
 stock_check_cache_proportion: 10
+filter_root_category: true
diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/ConductorCategoryManager.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/ConductorCategoryManager.php
index 3f40307..aa94e8b 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/ConductorCategoryManager.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/ConductorCategoryManager.php
@@ -231,6 +231,13 @@ private function resetResults() {
    */
   private function syncCategory(array $categories, $parent = NULL) {
 
+    // Remove top level item (Default Category) from the categories, if its set
+    // in configuration and category is with no parent.
+    $filter_root_category = \Drupal::config('acq_commerce.conductor')->get('filter_root_category');
+    if ($filter_root_category && $parent == NULL) {
+      $categories = $categories[0]['children'];
+    }
+
     foreach ($categories as $category) {
       if (!isset($category['category_id']) || !isset($category['name'])) {
         $this->logger->error('Invalid / missing category ID or name.');
diff --git a/docroot/modules/commerce/acq_commerce/src/Form/ConductorSettingsForm.php b/docroot/modules/commerce/acq_commerce/src/Form/ConductorSettingsForm.php
index bea6a7e..83f6553 100644
--- a/docroot/modules/commerce/acq_commerce/src/Form/ConductorSettingsForm.php
+++ b/docroot/modules/commerce/acq_commerce/src/Form/ConductorSettingsForm.php
@@ -41,6 +41,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       ->set('timeout', (int) $form_state->getValue('timeout'))
       ->set('verify_ssl', (bool) $form_state->getValue('verify_ssl'))
       ->set('product_page_size', (int) $form_state->getValue('page_size'))
+      ->set('filter_root_category', (bool) $form_state->getValue('filter_root_category'))
       ->save();
 
     return parent::submitForm($form, $form_state);
@@ -104,6 +105,12 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#default_value' => $config->get('product_page_size'),
     ];
 
+    $form['filter_root_category'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Filter root level category'),
+      '#default_value' => $config->get('filter_root_category'),
+    ];
+
     return parent::buildForm($form, $form_state);
   }
 
