diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index 1bc7153..f2f3f10 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -153,6 +153,13 @@ public function post(array $products = []) {
         continue;
       }
 
+      // Don't import configurable SKU if it has no configurable options.
+      if ($product['type'] == 'configurable' && empty($product['extension']['configurable_product_options'])) {
+        $this->logger->error('Empty configurable options for SKU: @sku', ['@sku' => $product['sku']]);
+        $failed++;
+        continue;
+      }
+
       $query = $this->queryFactory->get('acq_sku')
         ->condition('sku', $product['sku']);
       $nids = $query->execute();
