diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index e973987..ed67d9d 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -487,7 +487,9 @@ public function getCart() {
   public function convertToCustomerCart(array $cart) {
     $this->cart->cart_id = $cart['cart_id'];
     $this->cart->customer_id = $cart['customer_id'];
-    $this->cart->customer_email = $cart['customer_email'];
+    if (!empty($cart['customer_email'])) {
+      $this->cart->customer_email = $cart['customer_email'];
+    }
   }
 
   /**
diff --git a/modules/acq_cart/src/CartSessionStorage.php b/modules/acq_cart/src/CartSessionStorage.php
index eed6f82..46e9569 100644
--- a/modules/acq_cart/src/CartSessionStorage.php
+++ b/modules/acq_cart/src/CartSessionStorage.php
@@ -219,7 +219,7 @@ public function createCart() {
   /**
    * {@inheritdoc}
    */
-  public function associateCart($customer_id) {
+  public function associateCart($customer_id, $customer_email = '') {
     // We first update the session cart.
     $cart = $this->session->get(self::STORAGE_KEY);
     if (!$cart) {
@@ -229,6 +229,7 @@ public function associateCart($customer_id) {
     $data = [
       'cart_id' => $cart->id(),
       'customer_id' => $customer_id,
+      'customer_email' => $customer_email
     ];
     $cart->convertToCustomerCart($data);
     $this->session->set(self::STORAGE_KEY, $cart);
diff --git a/modules/acq_cart/src/CartStorageInterface.php b/modules/acq_cart/src/CartStorageInterface.php
index 45ff946..44e7519 100644
--- a/modules/acq_cart/src/CartStorageInterface.php
+++ b/modules/acq_cart/src/CartStorageInterface.php
@@ -54,7 +54,12 @@ public function createCart();
 
   /**
    * Associate the current cart in storage with a given customer.
+   *
+   * @param int $customer_id
+   *   Customer ID.
+   * @param string $customer_email
+   *   Customer E-Mail.
    */
-  public function associateCart($customer_id);
+  public function associateCart($customer_id, $customer_email);
 
 }
