diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 141789c..9ca1074 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -129,7 +129,7 @@ public function placeOrder($cart_id) {
     $result = [];
 
     try {
-      $result = $this->tryAgentRequest($doReq, 'placeOrder', 'cart');
+      $result = $this->tryAgentRequest($doReq, 'placeOrder');
     }
     catch (ConductorException $e) {
       throw new \Exception($e->getMessage(), $e->getCode());
