diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index 1c95645..8f04b8e 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -426,6 +426,7 @@ public function getCart() {
    */
   public function convertToCustomerCart(array $cart) {
     $this->cart->cart_id = $cart['cart_id'];
+    $this->cart->customer_id = $cart['customer_id'];
   }
 
   /**
diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 141789c..9ca1074 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -129,7 +129,7 @@ public function placeOrder($cart_id) {
     $result = [];
 
     try {
-      $result = $this->tryAgentRequest($doReq, 'placeOrder', 'cart');
+      $result = $this->tryAgentRequest($doReq, 'placeOrder');
     }
     catch (ConductorException $e) {
       throw new \Exception($e->getMessage(), $e->getCode());
