diff --git a/src/Form/ConductorSettingsForm.php b/src/Form/ConductorSettingsForm.php
index 359d6aee5..59527b94a 100644
--- a/src/Form/ConductorSettingsForm.php
+++ b/src/Form/ConductorSettingsForm.php
@@ -96,17 +96,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     ];
 
     $form['security']['hmac_id'] = [
-      '#type' => 'textfield',
+      '#type' => 'password',
       '#title' => $this->t('HMAC Key ID'),
       '#required' => TRUE,
-      '#default_value' => $config->get('hmac_id'),
     ];
 
     $form['security']['hmac_secret'] = [
-      '#type' => 'textfield',
+      '#type' => 'password',
       '#title' => $this->t('HMAC Key Secret'),
       '#required' => TRUE,
-      '#default_value' => $config->get('hmac_secret'),
     ];
 
     $form['page_size'] = [
