diff --git a/modules/acq_customer/acq_customer.permissions.yml b/modules/acq_customer/acq_customer.permissions.yml
new file mode 100644
index 0000000..857bdc0
--- /dev/null
+++ b/modules/acq_customer/acq_customer.permissions.yml
@@ -0,0 +1,6 @@
+'access own orders':
+  title: 'View own order details and list'
+
+'access all orders':
+  title: 'View all order details and list'
+  restrict access: true
diff --git a/modules/acq_customer/src/Controller/CustomerController.php b/modules/acq_customer/src/Controller/CustomerController.php
index 8b0644d..620cec2 100644
--- a/modules/acq_customer/src/Controller/CustomerController.php
+++ b/modules/acq_customer/src/Controller/CustomerController.php
@@ -45,7 +45,26 @@ public function ordersPage(UserInterface $user = NULL) {
   /**
    * Checks if user has access to the orders display page.
    */
-  public function checkAccess() {
-    return AccessResult::allowed()->cachePerPermissions();
+  public function checkAccess(UserInterface $user = NULL) {
+    // Something fishy, it should probably return 404 and not even reach here.
+    if (empty($user)) {
+      return AccessResult::forbidden();
+    }
+
+    // Load the current logged in user details.
+    $currentUser = \Drupal::currentUser();
+
+    // By design, only logged in users will be able to access the orders page.
+    if ($currentUser->isAnonymous()) {
+      return AccessResult::forbidden();
+    }
+
+    // If user is trying to access another user's orders, check admin perm.
+    if ($currentUser->id() != $user->id()) {
+      return AccessResult::allowedIfHasPermission($currentUser, 'access all orders');
+    }
+
+    // Check if user has access to view own orders.
+    return AccessResult::allowedIfHasPermission($currentUser, 'access own orders');
   }
 }
