diff --git a/modules/acq_sku_position/acq_sku_position.drush.inc b/modules/acq_sku_position/acq_sku_position.drush.inc
index d09fbc0be..fe63bb646 100644
--- a/modules/acq_sku_position/acq_sku_position.drush.inc
+++ b/modules/acq_sku_position/acq_sku_position.drush.inc
@@ -6,6 +6,7 @@
  */
 
 use Drupal\taxonomy\Entity\Term;
+use Drupal\Core\Cache\Cache;
 
 /**
  * Implements hook_drush_command().
@@ -112,5 +113,9 @@ function drush_acq_sku_position_position_sync($position_type = 'position') {
     }
   }
 
+  // Invalidate the cache tag for product index.
+  $tags = ["search_api_list:product"];
+  Cache::invalidateTags($tags);
+
   drush_print(dt('Product position sync completed!'));
 }
diff --git a/modules/acq_sku_position/acq_sku_position.api.php b/modules/acq_sku_position/acq_sku_position.api.php
new file mode 100644
index 000000000..3df390bee
--- /dev/null
+++ b/modules/acq_sku_position/acq_sku_position.api.php
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * @file
+ * Hooks specific to the acq_sku_position module.
+ */
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Allow other modules to take action after position sync finished.
+ */
+function hook_acq_sku_position_sync_finished() {
+
+}
+
+/**
+ * @} End of "addtogroup hooks".
+ */
diff --git a/modules/acq_sku_position/acq_sku_position.drush.inc b/modules/acq_sku_position/acq_sku_position.drush.inc
index fe63bb646..653e85c73 100644
--- a/modules/acq_sku_position/acq_sku_position.drush.inc
+++ b/modules/acq_sku_position/acq_sku_position.drush.inc
@@ -6,7 +6,6 @@
  */
 
 use Drupal\taxonomy\Entity\Term;
-use Drupal\Core\Cache\Cache;
 
 /**
  * Implements hook_drush_command().
@@ -113,9 +112,8 @@ function drush_acq_sku_position_position_sync($position_type = 'position') {
     }
   }
 
-  // Invalidate the cache tag for product index.
-  $tags = ["search_api_list:product"];
-  Cache::invalidateTags($tags);
+  // Allow other modules to take action after position sync finished.
+  \Drupal::moduleHandler()->invokeAll('acq_sku_position_sync_finished');
 
   drush_print(dt('Product position sync completed!'));
 }
