diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartInterface.php b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartInterface.php
index eb4bc4e..cf85f78 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartInterface.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_cart/src/CartInterface.php
@@ -65,6 +65,14 @@ public function items();
   public function addItemToCart($sku, $quantity);
 
   /**
+   * Removes an SKU from cart.
+   *
+   * @param string $sku
+   *   The name of the SKU to be removed from the cart.
+   */
+  public function removeItemFromCart($sku);
+
+  /**
    * Adds a raw item to the cart. Does not preform any validation
    * before appending to the cart items. Used for more complex items.
    *
