diff --git a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
index d548bf85f..33e534497 100644
--- a/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
+++ b/docroot/modules/commerce/acq_commerce/modules/acq_sku/src/Entity/SKU.php
@@ -248,12 +248,22 @@ public function getPluginInstance() {
    *   Found SKU
    */
   public static function loadFromSku($sku, $langcode = '', $log_not_found = TRUE, $create_translation = FALSE) {
+    $skus_static_cache = &drupal_static(__FUNCTION__, []);
+
     $is_multilingual = \Drupal::languageManager()->isMultilingual();
 
     if ($is_multilingual && empty($langcode)) {
       $langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
     }
 
+    $static_cache_sku_identifier = $sku . ':' . $langcode;
+
+    // Check if data is available in static cache, return from there.
+    if (isset($skus_static_cache[$static_cache_sku_identifier])) {
+      return $skus_static_cache[$static_cache_sku_identifier];
+    }
+
     $storage = \Drupal::entityTypeManager()->getStorage('acq_sku');
     $skus = $storage->loadByProperties(['sku' => $sku]);
 
@@ -298,6 +308,9 @@ public static function loadFromSku($sku, $langcode = '', $log_not_found = TRUE,
       }
     }
 
+    // Set value in static variable.
+    $skus_static_cache[$static_cache_sku_identifier] = $sku_entity;
+
     return $sku_entity;
   }
 
