--- a/js/soft-limit.js
+++ b/js/soft-limit.js
@@ -43,11 +43,11 @@

     // Hide facets over the limit.
     facetsList.each(function() {
-      $(this).children('li:gt(' + zero_based_limit + ')').once().hide();
+      $(this).children('li:gt(' + zero_based_limit + ')').once('apply-soft-limit').hide();
     });
 
     // Add "Show more" / "Show less" links.
-    facetsList.once().filter(function () {
+    facetsList.once('apply-soft-limit').filter(function () {
       return $(this).find('li').length > limit;
     }).each(function () {
       var facet = $(this);
