diff --git a/src/Robo/Commands/Deploy/DeployCommand.php b/src/Robo/Commands/Deploy/DeployCommand.php
index 487614cf3..a88439cdb 100644
--- a/src/Robo/Commands/Deploy/DeployCommand.php
+++ b/src/Robo/Commands/Deploy/DeployCommand.php
@@ -497,13 +497,16 @@ protected function setMultisiteFilePermissions($perms) {
    */
   protected function commit() {
     $this->say("Committing artifact to <comment>{$this->branchName}</comment>...");
-    $this->taskExecStack()
+    $result = $this->taskExecStack()
       ->dir($this->deployDir)
       ->exec("git add -A")
       ->exec("git commit --quiet -m '{$this->commitMessage}'")
-      ->stopOnFail()
       ->setVerbosityThreshold(VerbosityThresholdInterface::VERBOSITY_VERBOSE)
       ->run();
+
+    if (!$result->wasSuccessful()) {
+      throw new BltException("Failed to commit deployment artifact!");
+    }
   }

   /**
@@ -519,13 +522,16 @@ protected function push($identifier, $options) {
     }

     $task = $this->taskExecStack()
-      ->dir($this->deployDir)
-      ->stopOnFail();
+      ->dir($this->deployDir);
     foreach ($this->getConfigValue('git.remotes') as $remote) {
       $remote_name = md5($remote);
       $task->exec("git push $remote_name $identifier");
     }
-    $task->run();
+    $result = $task->run();
+
+    if (!$result->wasSuccessful()) {
+      throw new BltException("Failed to push deployment artifact!");
+    }
   }

   /**