diff --git a/acq_commerce.install b/acq_commerce.install
new file mode 100644
index 000000000..681a7bd34
--- /dev/null
+++ b/acq_commerce.install
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @file
+ * Install file.
+ */
+
+/**
+ * Implements hook_update_N().
+ *
+ * Add new role for sync user.
+ */
+function acq_commerce_update_8001() {
+  // Add the new role.
+  alshaya_config_install_configs(['user.role.acq_commerce_sync'], 'acq_commerce');
+}
diff --git a/config/install/user.role.acq_commerce_sync.yml b/config/install/user.role.acq_commerce_sync.yml
new file mode 100644
index 000000000..4f9704e07
--- /dev/null
+++ b/config/install/user.role.acq_commerce_sync.yml
@@ -0,0 +1,12 @@
+langcode: en
+status: true
+dependencies: {  }
+id: acq_commerce_sync
+label: ACQ Commerce Sync
+weight: 15
+is_admin: null
+permissions:
+  - 'restful post acq_categorysync'
+  - 'restful post acq_customer_delete'
+  - 'restful post acq_productstocksync'
+  - 'restful post acq_productsync'
diff --git a/acq_commerce.install b/acq_commerce.install
index 681a7bd34..3072e79b1 100644
--- a/acq_commerce.install
+++ b/acq_commerce.install
@@ -11,6 +11,11 @@
  * Add new role for sync user.
  */
 function acq_commerce_update_8001() {
+  $config_yaml = Yaml::parse(file_get_contents(drupal_get_path('module', 'acq_commerce') . '/config/install/user.role.acq_commerce_sync.yml'));
+
   // Add the new role.
-  alshaya_config_install_configs(['user.role.acq_commerce_sync'], 'acq_commerce');
+  \Drupal::configFactory()
+    ->getEditable('user.role.acq_commerce_sync')
+    ->setData($config_yaml)
+    ->save();
 }
diff --git a/acq_commerce.install b/acq_commerce.install
index 3072e79b1..59791ab4d 100644
--- a/acq_commerce.install
+++ b/acq_commerce.install
@@ -5,6 +5,8 @@
  * Install file.
  */
 
+use Symfony\Component\Yaml\Yaml;
+
 /**
  * Implements hook_update_N().
  *
