diff --git a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index 3aa9c8165..8dc7399a7 100644
--- a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -72,22 +72,25 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
         foreach ($tids as $tid => $values) {
           $sorted_options[$values->field_sku_option_id_value] = $options[$values->field_sku_option_id_value];
         }
-      }
 
-      $form['ajax']['configurables'][$attribute_code] = [
-        '#type' => 'select',
-        '#title' => $configurable['label'],
-        '#options' => $sorted_options,
-        '#required' => TRUE,
-        '#ajax' => [
-          'callback' => [$this, 'configurableAjaxCallback'],
-          'progress' => [
-            'type' => 'throbber',
-            'message' => NULL,
+        $form['ajax']['configurables'][$attribute_code] = [
+          '#type' => 'select',
+          '#title' => $configurable['label'],
+          '#options' => $sorted_options,
+          '#required' => TRUE,
+          '#ajax' => [
+            'callback' => [$this, 'configurableAjaxCallback'],
+            'progress' => [
+              'type' => 'throbber',
+              'message' => NULL,
+            ],
+            'wrapper' => 'configurable_ajax',
           ],
-          'wrapper' => 'configurable_ajax',
-        ],
-      ];
+        ];
+      }
+      else {
+        \Drupal::logger('acq_sku')->info('Product with sku: @sku seems to be configurable without any config options.', ['@sku' => $sku->getSku()]);
+      }
     }
 
     $form['sku_id'] = [
diff --git a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index 8dc7399a7..cf27ba58a 100644
--- a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -49,6 +49,7 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
       $attribute_code = $configurable['code'];
 
       $options = [];
+      $sorted_options = [];
 
       foreach ($configurable['values'] as $value) {
         $options[$value['value_id']] = $value['label'];
@@ -68,29 +69,28 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
         $query->orderBy('weight', 'ASC');
         $tids = $query->execute()->fetchAllAssoc('tid');
 
-        $sorted_options = [];
         foreach ($tids as $tid => $values) {
           $sorted_options[$values->field_sku_option_id_value] = $options[$values->field_sku_option_id_value];
         }
-
-        $form['ajax']['configurables'][$attribute_code] = [
-          '#type' => 'select',
-          '#title' => $configurable['label'],
-          '#options' => $sorted_options,
-          '#required' => TRUE,
-          '#ajax' => [
-            'callback' => [$this, 'configurableAjaxCallback'],
-            'progress' => [
-              'type' => 'throbber',
-              'message' => NULL,
-            ],
-            'wrapper' => 'configurable_ajax',
-          ],
-        ];
       }
       else {
         \Drupal::logger('acq_sku')->info('Product with sku: @sku seems to be configurable without any config options.', ['@sku' => $sku->getSku()]);
       }
+
+      $form['ajax']['configurables'][$attribute_code] = [
+        '#type' => 'select',
+        '#title' => $configurable['label'],
+        '#options' => $sorted_options,
+        '#required' => TRUE,
+        '#ajax' => [
+          'callback' => [$this, 'configurableAjaxCallback'],
+          'progress' => [
+            'type' => 'throbber',
+            'message' => NULL,
+          ],
+          'wrapper' => 'configurable_ajax',
+        ],
+      ];
     }
 
     $form['sku_id'] = [
