diff --git a/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php b/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
index aef8e913..e4b5e486 100644
--- a/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
+++ b/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionAttachQueue.php
@@ -57,7 +57,7 @@ public function processItem($data) {
           $update_sku_flag = TRUE;
         }

-        if ((isset($sku['final_price'])) && ($sku_entitys->final_price->value !== $sku['final_price'])) {
+        if ((isset($sku['final_price'])) && ($sku_entity->final_price->value !== $sku['final_price'])) {
           $sku_entity->final_price->value = $sku['final_price'];
           $update_sku_flag = TRUE;
         }
diff --git a/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php b/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
index 6244eba5..d62506a4 100644
--- a/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
+++ b/modules/acq_promotion/src/Plugin/QueueWorker/AcqPromotionDetachQueue.php
@@ -58,22 +58,21 @@ public function processItem($data) {
       $sku_entity->save();
     }

-    $sku_query_string = implode(',', $skus);
-
-    $endpoint = $this->apiVersion . '/ingest/product/sync?skus=' . $sku_query_string;
+    $sku_texts = implode(',', $skus);
+    $endpoint = $this->apiVersion . '/ingest/product/sync';

     $doReq = function ($client, $opt) use ($endpoint) {
       return $client->post($endpoint, $opt);
     };

     try {
-      $this->tryIngestRequest($doReq, 'productFullSync', 'products');
+      $this->tryIngestRequest($doReq, 'productFullSync', 'products', $sku_texts);
     }
     catch (ConductorException $e) {
     }

     $this->loggerFactory->get('acq_sku')->info('Detached Promotion:@promo from SKUs: @skus',
-      ['@promo' => $promotion_nid, '@skus' => implode(',', $skus)]);
+      ['@promo' => $promotion_nid, '@skus' => $sku_texts]);
   }

 }
diff --git a/modules/acq_sku/acq_sku.drush.inc b/modules/acq_sku/acq_sku.drush.inc
index 572ce24e..5d1926d6 100644
--- a/modules/acq_sku/acq_sku.drush.inc
+++ b/modules/acq_sku/acq_sku.drush.inc
@@ -26,6 +26,10 @@ function acq_sku_drush_command() {
     'aliases'     => ['acsp'],
     'examples'    => [
       'drush acsp' => 'Run a full product synchronization of all available products.',
+      'drush acsp --skus=\'M-H3495 130 2  FW\',\'M-H3496 130 004FW\',\'M-H3496 130 005FW\'' => 'Synchronize sku data for the skus M-H3495 130 2  FW, M-H3496 130 004FW & M-H3496 130 005FW only.',
+    ],
+    'options'     => [
+      'skus' => 'SKUs to import (like query).',
     ],
   ];

@@ -54,7 +58,9 @@ function acq_sku_drush_command() {
 function drush_acq_sku_sync_commerce_products() {
   drush_print(dt('Synchronizing all commerce products, this usually takes some time...'));
   $container = \Drupal::getContainer();
-  $container->get('acq_commerce.ingest_api')->productFullSync();
+  $skus = drush_get_option('skus');
+
+  $container->get('acq_commerce.ingest_api')->productFullSync($skus);
 }

 /**
diff --git a/src/Conductor/IngestAPIWrapper.php b/src/Conductor/IngestAPIWrapper.php
index d0a53bbb..a70fae1a 100644
--- a/src/Conductor/IngestAPIWrapper.php
+++ b/src/Conductor/IngestAPIWrapper.php
@@ -34,7 +34,7 @@ public function __construct(ClientFactory $client_factory, ConfigFactoryInterfac
   /**
    * Performs full conductor sync.
    */
-  public function productFullSync() {
+  public function productFullSync($skus = '') {
     foreach (acq_commerce_get_store_language_mapping() as $langcode => $store_id) {
       if (empty($store_id)) {
         continue;
@@ -54,7 +54,7 @@ public function productFullSync() {
       };

       try {
-        $this->tryIngestRequest($doReq, 'productFullSync', 'products');
+        $this->tryIngestRequest($doReq, 'productFullSync', 'products', $skus);
       }
       catch (ConductorException $e) {
       }
diff --git a/src/Conductor/IngestRequestTrait.php b/src/Conductor/IngestRequestTrait.php
index 5e1756fc..bea63e25 100644
--- a/src/Conductor/IngestRequestTrait.php
+++ b/src/Conductor/IngestRequestTrait.php
@@ -47,13 +47,20 @@
    *   Action name for logging.
    * @param string $reskey
    *   Result data key (or NULL)
+   * @param string $skus
+   *   List of skus that need to be synced.
    *
    * @throws ConductorException
    */
-  protected function tryIngestRequest(callable $doReq, $action, $reskey = NULL) {
+  protected function tryIngestRequest(callable $doReq, $action, $reskey = NULL, $skus = '') {

     $client = $this->clientFactory->createClient();
     $reqOpts = [];
+
+    if (!empty($skus)) {
+      $reqOpts['query']['skus'] = $skus;
+    }
+
     $logger = ($this->logger) ?: \Drupal::logger('acq_commerce');

     if ($this->debug) {
diff --git a/src/Conductor/IngestAPIWrapper.php b/src/Conductor/IngestAPIWrapper.php
index a70fae1a2..81adcfda1 100644
--- a/src/Conductor/IngestAPIWrapper.php
+++ b/src/Conductor/IngestAPIWrapper.php
@@ -47,14 +47,20 @@ public function productFullSync($skus = '') {
         fclose($fp);
       }

-      $endpoint = $this->apiVersion . '/ingest/product/sync?store_id=' . $store_id;
+      $endpoint = $this->apiVersion . '/ingest/product/sync';
+
+      $doReq = function ($client, $opt) use ($endpoint, $store_id, $skus) {
+        if (!empty($skus)) {
+          $opt['query']['skus'] = $skus;
+        }
+
+        $opt['query']['store_id'] = $store_id;

-      $doReq = function ($client, $opt) use ($endpoint) {
         return $client->post($endpoint, $opt);
       };

       try {
-        $this->tryIngestRequest($doReq, 'productFullSync', 'products', $skus);
+        $this->tryIngestRequest($doReq, 'productFullSync', 'products');
       }
       catch (ConductorException $e) {
       }
diff --git a/src/Conductor/IngestRequestTrait.php b/src/Conductor/IngestRequestTrait.php
index bea63e250..31f67809e 100644
--- a/src/Conductor/IngestRequestTrait.php
+++ b/src/Conductor/IngestRequestTrait.php
@@ -47,20 +47,14 @@
    *   Action name for logging.
    * @param string $reskey
    *   Result data key (or NULL)
-   * @param string $skus
-   *   List of skus that need to be synced.
    *
    * @throws ConductorException
    */
-  protected function tryIngestRequest(callable $doReq, $action, $reskey = NULL, $skus = '') {
+  protected function tryIngestRequest(callable $doReq, $action, $reskey = NULL) {

     $client = $this->clientFactory->createClient();
     $reqOpts = [];

-    if (!empty($skus)) {
-      $reqOpts['query']['skus'] = $skus;
-    }
-
     $logger = ($this->logger) ?: \Drupal::logger('acq_commerce');

     if ($this->debug) {
