diff --git a/acq_commerce.module b/acq_commerce.module
index 6aec402..809cc63 100644
--- a/acq_commerce.module
+++ b/acq_commerce.module
@@ -16,3 +16,40 @@ function acq_commerce_page_attachments(array &$attachments) {
     $attachments['#attached']['library'][] = 'acq_commerce/drupal.acq_commerce.toolbar';
   }
 }
+
+/**
+ * Implements hook_theme().
+ */
+function acq_commerce_theme($existing, $type, $theme, $path) {
+  return [
+    'acq_commerce_price' => [
+      'variables' => [
+        'price' => '',
+        'currency_format' => '',
+        'currency_code_position' => '',
+      ],
+    ],
+  ];
+}
+
+/**
+ * Format price value with currency.
+ *
+ * Default template: acq-commerce-price.html.twig.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - price: a price value.
+ */
+function template_preprocess_acq_commerce_price(array &$variables) {
+  // Fetch the config.
+  $config = \Drupal::configFactory()->get('acq_commerce.currency');
+
+  // Fetch the currency format from the config factor.
+  $variables['currency_format'] = $config->get('currency_code');
+
+  // Fetch the currency code position.
+  $variables['currency_code_position'] = $config->get('currency_code_position');
+
+  $variables['price'] = number_format($variables['price'], $config->get('decimal_points'));
+}
diff --git a/config/install/acq_commerce.currency.yml b/config/install/acq_commerce.currency.yml
index 85b3f3b..a603eb2 100644
--- a/config/install/acq_commerce.currency.yml
+++ b/config/install/acq_commerce.currency.yml
@@ -1,3 +1,4 @@
 currency_code: '$'
 currency_code_position: 'before'
+decimal_points: 2
 langcode: en
diff --git a/modules/acq_cart/acq_cart.module b/modules/acq_cart/acq_cart.module
index b3dca5c..0a8d5fe 100644
--- a/modules/acq_cart/acq_cart.module
+++ b/modules/acq_cart/acq_cart.module
@@ -34,8 +34,6 @@ function acq_cart_theme($existing, $type, $theme, $path) {
       'variables' => [
         'quantity' => 0,
         'total' => NULL,
-        'currency_format' => $currency_format,
-        'currency_code_position' => $currency_code_position,
       ],
       'template' => 'block--acqminicart',
     ],
diff --git a/modules/acq_cart/src/Controller/GetMiniCart.php b/modules/acq_cart/src/Controller/GetMiniCart.php
index 8faa6ed..7c56736 100644
--- a/modules/acq_cart/src/Controller/GetMiniCart.php
+++ b/modules/acq_cart/src/Controller/GetMiniCart.php
@@ -6,6 +6,7 @@
 use Drupal\acq_cart\CartStorageInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
 
 /**
  * Class GetMiniCart.
@@ -53,29 +54,23 @@ public function content() {
 
     $totals = $cart->totals();
 
-    // Fetch the config.
-    $config = \Drupal::configFactory()
-      ->get('acq_commerce.currency');
-
-    // Fetch the currency format from the config factor.
-    $currency_format = $config->get('currency_code');
-
-    // Fetch the currency code position.
-    $currency_code_position = $config->get('currency_code_position');
-
     // The grand total including discounts and taxes.
     $grand_total = $totals['grand'] < 0 || $totals['grand'] == NULL ? 0 : $totals['grand'];
+    $total = [
+      '#theme' => 'acq_commerce_price',
+      '#price' => $grand_total,
+    ];
 
     // Use the template to render the HTML.
     $output = [
       '#theme' => 'acq_cart_mini_cart',
       '#quantity' => $cart->getCartItemsCount(),
-      '#total' => $grand_total,
-      '#currency_format' => $currency_format,
-      '#currency_code_position' => $currency_code_position,
+      '#total' => render($total),
     ];
 
-    return new JsonResponse(drupal_render($output)->jsonSerialize());
+    $response = new Response();
+    $response->setContent(render($output));
+    return $response;
   }
 
 }
diff --git a/modules/acq_cart/templates/block--acqminicart.html.twig b/modules/acq_cart/templates/block--acqminicart.html.twig
index 8afcf9f..5216a9e 100644
--- a/modules/acq_cart/templates/block--acqminicart.html.twig
+++ b/modules/acq_cart/templates/block--acqminicart.html.twig
@@ -1,8 +1,4 @@
 <div class="acq-mini-cart">
-  {% if currency_code_position == 'before' %}
-    <span class="total"><span class="prefix">{{ currency_format }}</span><span class="total--amount">{{ total }}</span></span>
-  {% else %}
-    <span class="total">{{ total }}<span class="prefix">{{ currency_format }}</span></span>
-  {% endif %}
+  {{ total }}
   <a class="cart-link" href="{{ url('acq_cart.cart') }}"><span class="quantity">{{ quantity }}</span></a>
 </div>
diff --git a/src/Form/CurrencySettingsForm.php b/src/Form/CurrencySettingsForm.php
index c05da27..30157b2 100644
--- a/src/Form/CurrencySettingsForm.php
+++ b/src/Form/CurrencySettingsForm.php
@@ -31,10 +31,10 @@ public function getEditableConfigNames() {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-
     $this->config('acq_commerce.currency')
       ->set('currency_code', $form_state->getValue('currency_code'))
       ->set('currency_code_position', $form_state->getValue('currency_code_position'))
+      ->set('decimal_points', $form_state->getValue('decimal_points'))
       ->save();
 
     return parent::submitForm($form, $form_state);
@@ -66,6 +66,14 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#required' => TRUE,
     ];
 
+    $form['decimal_points'] = [
+      '#type' => 'select',
+      '#title' => t('Decimal points'),
+      '#default_value' => $config->get('decimal_points'),
+      '#options' => [1 => 1, 2 => 2, 3 => 3, 4 => 4, 5 => 5],
+      '#required' => TRUE,
+    ];
+
     return parent::buildForm($form, $form_state);
   }
 
diff --git a/templates/acq-commerce-price.html.twig b/templates/acq-commerce-price.html.twig
new file mode 100644
index 0000000..d842130
--- /dev/null
+++ b/templates/acq-commerce-price.html.twig
@@ -0,0 +1,13 @@
+{% if price > 0 %}
+  {% if currency_code_position == 'before' %}
+    <span class="price-wrapper">
+      <span class="price-currency prefix">{{ currency_format }}</span>
+      <span class="price-amount">{{ price }}</span>
+    </span>
+  {% else %}
+    <span class="price-wrapper">
+      <span class="price-amount">{{ price }}</span>
+      <span class="price-currency suffix">{{ currency_format }}</span>
+    </span>
+  {% endif %}
+{% endif %}
