diff --git a/modules/acq_checkout/src/ACQAddressFormatter.php b/modules/acq_checkout/src/ACQAddressFormatter.php
index 0381f33..b18087f 100644
--- a/modules/acq_checkout/src/ACQAddressFormatter.php
+++ b/modules/acq_checkout/src/ACQAddressFormatter.php
@@ -46,7 +46,7 @@ public function render($address) {
    *   A renderable array.
    */
   protected function viewElement($address, $langcode) {
-    $country_code = $address->country;
+    $country_code = $address->country_id;
     $country_repository = \Drupal::service('address.country_repository');
     $countries = $country_repository->getList();
     $address_format_repository = \Drupal::service('address.address_format_repository');
@@ -124,7 +124,7 @@ protected function getValues($address, $country_code, AddressFormat $address_for
         // This level is empty, so there can be no sublevels.
         break;
       }
-      $parents[] = $index ? $original_values[$subdivision_fields[$index - 1]] : $address->country;
+      $parents[] = $index ? $original_values[$subdivision_fields[$index - 1]] : $address->country_id;
       $subdivision_repository = \Drupal::service('address.subdivision_repository');
       $subdivision = $subdivision_repository->get($values[$field], $parents);
       if (!$subdivision) {
diff --git a/modules/acq_checkout/src/Plugin/CheckoutPane/AddressFormBase.php b/modules/acq_checkout/src/Plugin/CheckoutPane/AddressFormBase.php
index 17c6bd5..a921b7a 100644
--- a/modules/acq_checkout/src/Plugin/CheckoutPane/AddressFormBase.php
+++ b/modules/acq_checkout/src/Plugin/CheckoutPane/AddressFormBase.php
@@ -15,7 +15,7 @@ class AddressFormBase extends CheckoutPaneBase implements CheckoutPaneInterface
    */
   public function buildPaneForm(array $pane_form, FormStateInterface $form_state, array &$complete_form) {
     $address = $form_state->getTemporaryValue('address');
-    $country = isset($address->country) ? $address->country : 'US';
+    $country = isset($address->country_id) ? $address->country_id : 'US';
     $countryRepository = \Drupal::service('address.country_repository');
     $subdivisionRepository = \Drupal::service('address.subdivision_repository');
     $addressFormatRepository = \Drupal::service('address.address_format_repository');
@@ -51,10 +51,10 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
       '#required' => TRUE,
       '#placeholder' => $this->t('Address Line 1*'),
     ];
-    $pane_form['address']['phone'] = [
+    $pane_form['address']['telephone'] = [
       '#type' => 'textfield',
       '#title' => $this->t('Telephone'),
-      '#default_value' => isset($address->phone) ? $address->phone : '',
+      '#default_value' => isset($address->telephone) ? $address->telephone : '',
       '#required' => TRUE,
       '#placeholder' => $this->t('Telephone'),
     ];
@@ -94,7 +94,7 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
       '#required' => TRUE,
       '#placeholder' => $labels['postalCode'],
     ];
-    $pane_form['address']['dynamic_parts']['country'] = [
+    $pane_form['address']['dynamic_parts']['country_id'] = [
       '#type' => 'select',
       '#title' => $this->t('Country'),
       '#options' => $countryRepository->getList(),
@@ -116,7 +116,7 @@ public static function addressAjaxCallback($form, FormStateInterface $form_state
     $pane_id = $form_state->getTemporaryValue('pane_id');
     $dynamic_parts =& $form[$pane_id]['address']['dynamic_parts'];
     $values = $form_state->getValue($form['#parents']);
-    $country = $values[$pane_id]['address']['dynamic_parts']['country'];
+    $country = $values[$pane_id]['address']['dynamic_parts']['country_id'];
     $countryRepository = \Drupal::service('address.country_repository');
     $addressFormatRepository = \Drupal::service('address.address_format_repository');
     $address_format = $addressFormatRepository->get($country);
diff --git a/modules/acq_checkout/src/Plugin/CheckoutPane/BillingInformation.php b/modules/acq_checkout/src/Plugin/CheckoutPane/BillingInformation.php
index 865a82b..58dd4a1 100644
--- a/modules/acq_checkout/src/Plugin/CheckoutPane/BillingInformation.php
+++ b/modules/acq_checkout/src/Plugin/CheckoutPane/BillingInformation.php
@@ -75,7 +75,7 @@ public static function updateAddressAjaxCallback($form, FormStateInterface $form
     $field_names = [
       'first_name',
       'last_name',
-      'phone',
+      'telephone',
       'street',
       'street2',
     ];
@@ -84,7 +84,7 @@ public static function updateAddressAjaxCallback($form, FormStateInterface $form
       'city',
       'region',
       'postcode',
-      'country',
+      'country_id',
     ];
 
     foreach ($field_names as $field_name) {
diff --git a/modules/acq_checkout/src/Plugin/CheckoutPane/ShippingInformation.php b/modules/acq_checkout/src/Plugin/CheckoutPane/ShippingInformation.php
index 8e6cd97..03fc3db 100644
--- a/modules/acq_checkout/src/Plugin/CheckoutPane/ShippingInformation.php
+++ b/modules/acq_checkout/src/Plugin/CheckoutPane/ShippingInformation.php
@@ -123,7 +123,7 @@ public static function updateAddressAjaxCallback($form, FormStateInterface $form
     $field_names = [
       'first_name' => 'firstname',
       'last_name' => 'lastname',
-      'phone' => 'phone',
+      'telephone' => 'telephone',
       'street' => 'street',
       'street2' => 'street2',
     ];
@@ -132,7 +132,7 @@ public static function updateAddressAjaxCallback($form, FormStateInterface $form
       'city',
       'region',
       'postcode',
-      'country',
+      'country_id',
     ];
 
     foreach ($field_names as $field_key => $field_name) {
@@ -148,10 +148,10 @@ public static function updateAddressAjaxCallback($form, FormStateInterface $form
 
       $address = [
         'region' => $values['dynamic_parts']['region'],
-        'country' => $values['dynamic_parts']['country'],
+        'country_id' => $values['dynamic_parts']['country_id'],
         'street' => $values['street'],
         'street2' => $values['street2'],
-        'phone' => $values['phone'],
+        'telephone' => $values['telephone'],
         'postcode' => $values['dynamic_parts']['postcode'],
         'city' => $values['dynamic_parts']['city'],
         'firstname' => $values['first_name'],
