diff --git a/modules/acq_sku/acq_sku.module b/modules/acq_sku/acq_sku.module
index d9263b9..c0c96aa 100755
--- a/modules/acq_sku/acq_sku.module
+++ b/modules/acq_sku/acq_sku.module
@@ -37,3 +37,50 @@ function acq_sku_preprocess_acq_sku(&$variables) {
   }
 
 }
+
+/**
+ * Function to allow custom modules to add base fields for acq_sku entity type.
+ *
+ * @param array $fields
+ *   An associative array containing field_machine_name as key and following
+ *   data in value:
+ *   - parent: (Required) Parent key in the array where to look for data.
+ *   - label: (Required) Label to be used for admin forms and display.
+ *   - description: Description of the field to be used in admin forms.
+ *   - cardinality: Number of values allowed to be stored.
+ *   - type: Type of the field.
+ *   - weight: Default weight of the field in form and display.
+ *   - configurable: Whether the field is part of configurable options.
+ *   - visible_view: Whether the field should be visible while viewing content.
+ *   - visible_form: Whether the field should be visible in form.
+ *   - serialize: (Optional) Should the data be stored as serialized.
+ */
+function acq_sku_add_base_fields(array $fields) {
+  // Get the fields from config.
+  $configFactory = \Drupal::configFactory()->getEditable('acq_sku.base_field_additions');
+  $additionalFields = $configFactory->getRawData();
+
+  // Initialise with array if required.
+  $additionalFields = empty($additionalFields) || !is_array($additionalFields) ? [] : $additionalFields;
+
+  foreach ($fields as $machine_name => $field) {
+    // Confirm we have minimum required data available to create the fields.
+    if (empty($field['label'])) {
+      throw new \Exception('Field Label cannot be empty.');
+    }
+
+    // Check if the field is already available in config.
+    if (isset($additionalFields[$machine_name])) {
+      throw new \Exception('Field machine name used already.');
+    }
+
+    // Add the field to storedFields so we store them to config.
+    $additionalFields[$machine_name] = $field;
+  }
+
+  // Store the fields in config.
+  $configFactory->setData($additionalFields)->save();
+
+  // Apply entity updates.
+  \Drupal::service("entity.definition_update_manager")->applyUpdates();
+}
diff --git a/modules/acq_sku/src/Entity/SKU.php b/modules/acq_sku/src/Entity/SKU.php
index 0de4ab1..d106b41 100755
--- a/modules/acq_sku/src/Entity/SKU.php
+++ b/modules/acq_sku/src/Entity/SKU.php
@@ -315,6 +315,84 @@ class SKU extends ContentEntityBase implements SKUInterface {
       ->setLabel(t('Changed'))
       ->setDescription(t('The time that the entity was last edited.'));
 
+    // Get all the fields added by other modules and add them as base fields.
+    $additionalFields = \Drupal::config('acq_sku.base_field_additions')->getRawData();
+
+    // Get the default weight increment value from variables.
+    $defaultWeightIncrement = \Drupal::state()->get('acq_sku.base_field_weight_increment', 20);
+
+    // Check if we have additional fields to be added as base fields.
+    if (!empty($additionalFields) && is_array($additionalFields)) {
+      foreach ($additionalFields as $machine_name => $field_info) {
+        // Initialise the field variable.
+        $field = NULL;
+
+        // Showing the fields at the bottom.
+        $weight = $defaultWeightIncrement + count($fields);
+
+        switch ($field_info['type']) {
+          case 'string':
+            $field = BaseFieldDefinition::create('string');
+
+            if ($field_info['visible_view']) {
+              $field->setDisplayOptions('view', [
+                'label' => 'above',
+                'type' => 'string',
+                'weight' => $weight,
+              ]);
+            }
+
+            if ($field_info['visible_form']) {
+              $field->setDisplayOptions('form', [
+                'type' => 'string_textfield',
+                'weight' => $weight,
+              ]);
+            }
+            break;
+
+          case 'text_long':
+            $field = BaseFieldDefinition::create('text_long');
+
+            if ($field_info['visible_view']) {
+              $field->setDisplayOptions('view', [
+                'label' => 'hidden',
+                'type' => 'text_default',
+                'weight' => $weight,
+              ]);
+            }
+
+            if ($field_info['visible_form']) {
+              $field->setDisplayOptions('form', [
+                'type' => 'text_textfield',
+                'weight' => $weight,
+              ]);
+            }
+            break;
+        }
+
+        // Check if we don't have the field type defined yet.
+        if (empty($field)) {
+          throw new \RuntimeException('Field type not defined yet, please contact TA.');
+        }
+
+        $field->setLabel($field_info['label']);
+
+        // Update cardinality with default value if empty.
+        $field_info['description'] = empty($field_info['description']) ? 1 : $field_info['description'];
+        $field->setDescription($field_info['description']);
+
+        // Update cardinality with default value if empty.
+        $field_info['cardinality'] = empty($field_info['cardinality']) ? 1 : $field_info['cardinality'];
+        $field->setCardinality($field_info['cardinality']);
+
+        $field->setDisplayConfigurable('form', 1);
+        $field->setDisplayConfigurable('view', 1);
+
+        // We will use attr prefix to avoid conflicts with default base fields.
+        $fields['attr_' . $machine_name] = $field;
+      }
+    }
+
     return $fields;
   }
 
diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index 1ce1d0c..3212cc3 100755
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -3,6 +3,7 @@
 namespace Drupal\acq_sku\Plugin\rest\resource;
 
 use Drupal\acq_sku\CategoryRepositoryInterface;
+use Drupal\acq_sku\Entity\SKU;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Entity\Query\QueryFactory;
 use Drupal\Core\Config\ConfigFactoryInterface;
@@ -231,6 +232,12 @@ class ProductSyncResource extends ResourceBase {
         $created++;
       }
 
+      // Update the fields based on the values from attributes.
+      $this->updateAttributeFields($sku, $product['attributes']);
+
+      // Update the fields based on the values from extension.
+      $this->updateExtensionFields($sku, $product['extension']);
+
       $plugin_manager = \Drupal::service('plugin.manager.sku');
       $plugin_definition = $plugin_manager->pluginFromSKU($sku);
 
@@ -366,4 +373,84 @@ class ProductSyncResource extends ResourceBase {
     }
   }
 
+  /**
+   * Update attribute fields.
+   *
+   * Update the fields based on the values from attributes.
+   *
+   * @param Drupal\acq_sku\Entity\SKU $sku
+   *   The root SKU.
+   * @param array $attributes
+   *   The attributes to set.
+   */
+  private function updateAttributeFields(SKU $sku, array $attributes) {
+    $additionalFields = \Drupal::config('acq_sku.base_field_additions')->getRawData();
+
+    // Loop through all the attributes available for this particular SKU.
+    foreach ($attributes as $key => $value) {
+      // Check if attribute is required by us.
+      if (isset($additionalFields[$key])) {
+        $field = $additionalFields[$key];
+
+        if ($field['parent'] != 'attributes') {
+          continue;
+        }
+
+        $value = $field['cardinality'] != 1 ? explode(',', $value) : $value;
+        $field_key = 'attr_' . $key;
+
+        switch ($field['type']) {
+          case 'string':
+            $sku->{$field_key}->setValue($value);
+            break;
+
+          case 'text_long':
+            $value = isset($field['serialize']) ? serialize($value) : $value;
+            $sku->{$field_key}->setValue($value);
+            break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Update extension fields.
+   *
+   * Update the fields based on the values from extension.
+   *
+   * @param \Drupal\acq_sku\Entity\SKU $sku
+   *   The root SKU.
+   * @param array $attributes
+   *   The attributes to set.
+   */
+  private function updateExtensionFields(SKU $sku, array $attributes) {
+    $additionalFields = \Drupal::config('acq_sku.base_field_additions')->getRawData();
+
+    // Loop through all the attributes available for this particular SKU.
+    foreach ($attributes as $key => $value) {
+      // Check if attribute is required by us.
+      if (isset($additionalFields[$key])) {
+        $field = $additionalFields[$key];
+
+        if ($field['parent'] != 'extension') {
+          continue;
+        }
+
+        $field_key = 'attr_' . $key;
+
+        switch ($field['type']) {
+          case 'string':
+            $value = $field['cardinality'] != 1 ? explode(',', $value) : $value;
+            $sku->{$field_key}->setValue($value);
+            break;
+
+          case 'text_long':
+            $value = isset($field['serialize']) ? serialize($value) : $value;
+            $sku->{$field_key}->setValue($value);
+            break;
+        }
+      }
+    }
+  }
+
 }
