diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index eab01da6..39e36684 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -394,6 +394,19 @@ public function getShippingMethodAsString() {
   /**
    * {@inheritdoc}
    */
+  public function clearShippingMethod() {
+    // Unset the value set just now (before update cart).
+    unset($this->cart->carrier);
+
+    // Unset the value set in extension (updated cart response).
+    if (isset($this->cart, $this->cart->extension, $this->cart->extension['shipping_method'])) {
+      unset($this->cart->extension['shipping_method']);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function getPaymentMethod($full_details = TRUE) {
     if (!isset($this->cart, $this->cart->payment)) {
       return [];
diff --git a/modules/acq_cart/src/CartInterface.php b/modules/acq_cart/src/CartInterface.php
index e1a0e02a..9906e3a2 100644
--- a/modules/acq_cart/src/CartInterface.php
+++ b/modules/acq_cart/src/CartInterface.php
@@ -153,6 +153,11 @@ public function getShippingMethod();
   public function getShippingMethodAsString();
 
   /**
+   * Clears the shipping method info in cart.
+   */
+  public function clearShippingMethod();
+
+  /**
    * Gets the shipping method.
    *
    * @param string $carrier
diff --git a/modules/acq_cart/src/CartSessionStorage.php b/modules/acq_cart/src/CartSessionStorage.php
index 2151c1a5..fac22e05 100644
--- a/modules/acq_cart/src/CartSessionStorage.php
+++ b/modules/acq_cart/src/CartSessionStorage.php
@@ -126,6 +126,15 @@ public function clearCart() {
   /**
    * {@inheritdoc}
    */
+  public function clearShippingMethodSession() {
+    $cart = $this->getCart(FALSE);
+    $cart->clearShippingMethod();
+    $this->addCart($cart);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function getCart($create_new = TRUE) {
     $cart = $this->session->get(self::STORAGE_KEY);
 
diff --git a/modules/acq_cart/src/CartStorageInterface.php b/modules/acq_cart/src/CartStorageInterface.php
index 95f3f75f..8a6aad86 100644
--- a/modules/acq_cart/src/CartStorageInterface.php
+++ b/modules/acq_cart/src/CartStorageInterface.php
@@ -27,6 +27,11 @@ public function restoreCart($cart_id);
   public function clearCart();
 
   /**
+   * Clears the shipping method details in cart stored in session.
+   */
+  public function clearShippingMethodSession();
+
+  /**
    * Gets the current card ID.
    *
    * @param bool $create_new
