diff --git a/acq_commerce.api.php b/acq_commerce.api.php
new file mode 100644
index 0000000..958e794
--- /dev/null
+++ b/acq_commerce.api.php
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @file
+ * Hooks specific to the acq_commerce module.
+ */
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Alter customer data before sending to API.
+ *
+ * @param array $request_options
+ *   API request options to alter.
+ */
+function hook_acq_commerce_update_customer_api_request_alter(array &$request_options) {
+
+}
+
+/**
+ * @} End of "addtogroup hooks".
+ */
diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 5e37081..d3e6399 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -276,15 +276,17 @@ public function getPaymentMethods($cart_id) {
    *   Customer last name.
    * @param string $email
    *   Customer e-mail.
+   * @param string $password
+   *   Optional password.
    *
    * @return array
    *   New customer array.
    */
-  public function createCustomer($first_name, $last_name, $email) {
+  public function createCustomer($first_name, $last_name, $email, $password = NULL) {
     // First check if the user exists in Magento.
     try {
       if ($existingCustomer = $this->getCustomer($email)) {
-        return $this->updateCustomer($existingCustomer['customer_id'], $first_name, $last_name, $email);
+        return $this->updateCustomer($existingCustomer['customer_id'], $first_name, $last_name, $email, $password);
       }
     }
     catch (\Exception $e) {
@@ -292,7 +294,7 @@ public function createCustomer($first_name, $last_name, $email) {
       // already in magento.
     }
 
-    return $this->updateCustomer(NULL, $first_name, $last_name, $email);
+    return $this->updateCustomer(NULL, $first_name, $last_name, $email, $password);
   }
 
   /**
@@ -306,14 +308,16 @@ public function createCustomer($first_name, $last_name, $email) {
    *   Customer last name.
    * @param string $email
    *   Customer e-mail.
+   * @param string $password
+   *   Optional password.
    *
    * @return array
    *   New customer array.
    */
-  public function updateCustomer($customer_id, $first_name, $last_name, $email) {
+  public function updateCustomer($customer_id, $first_name, $last_name, $email, $password = NULL) {
     $endpoint = "customer";
 
-    $doReq = function ($client, $opt) use ($endpoint, $customer_id, $first_name, $last_name, $email) {
+    $doReq = function ($client, $opt) use ($endpoint, $customer_id, $first_name, $last_name, $email, $password) {
       if (!empty($customer_id)) {
         $opt['form_params']['customer[customer_id]'] = $customer_id;
       }
@@ -322,6 +326,13 @@ public function updateCustomer($customer_id, $first_name, $last_name, $email) {
       $opt['form_params']['customer[lastname]'] = $last_name;
       $opt['form_params']['customer[email]'] = $email;
 
+      if (!empty($password)) {
+        $opt['form_params']['password'] = $password;
+      }
+
+      // Invoke the alter hook to allow all modules to update the customer data.
+      \Drupal::moduleHandler()->alter('acq_commerce_update_customer_api_request', $opt);
+
       return ($client->post($endpoint, $opt));
     };
 
@@ -338,6 +349,38 @@ public function updateCustomer($customer_id, $first_name, $last_name, $email) {
   }
 
   /**
+   * Authenticate customer.
+   *
+   * @param string $email
+   *   Customer e-mail.
+   * @param string $password
+   *   Password.
+   *
+   * @return array
+   *   New customer array.
+   */
+  public function authenticateCustomer($email, $password) {
+    $endpoint = 'customer/' . $email;
+
+    $doReq = function ($client, $opt) use ($endpoint, $password) {
+      $opt['form_params']['password'] = $password;
+
+      return ($client->post($endpoint, $opt));
+    };
+
+    $customer = [];
+
+    try {
+      $customer = $this->tryAgentRequest($doReq, 'authenticateCustomer', 'customer');
+    }
+    catch (ConductorException $e) {
+      throw new \Exception($e->getMessage(), $e->getCode());
+    }
+
+    return $customer;
+  }
+
+  /**
    * Gets customer by email.
    *
    * @param string $email
