diff --git a/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php b/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
index 052dc4fa5..6a8bf715a 100644
--- a/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
+++ b/modules/acq_sku/src/AcquiaCommerce/SKUPluginBase.php
@@ -188,7 +188,12 @@ public function getDisplayNode(SKU $sku, $check_parent = TRUE, $create_translati
         return $node->addTranslation($sku->language()->getId());
       }
 
-      throw new \Exception(new FormattableMarkup('Node translation not found of @sku for @langcode', ['@sku' => $sku->id(), '@langcode' => $sku->language()->getId()]), 404);
+      // Just log the message and continue.
+      // Don't want to show any fatal error anywhere.
+      \Drupal::logger('acq_sku')->warning('Node translation not found of @sku for @langcode', [
+        '@sku' => $sku->getSku(),
+        '@langcode' => $sku->language()->getId(),
+      ]);
     }
 
     return $node;
