diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index 00a0d80..0ca499f 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -481,11 +481,11 @@ public function getCart() {
       $cart = $this->cart;
 
       // Don't set blank addresses, Magento doesn't like this.
-      if (isset($cart->shipping) && empty($cart->shipping->street)) {
+      if (isset($cart->shipping) && empty($cart->shipping->customer_address_id) && empty($cart->shipping->street)) {
         unset($cart->shipping);
       }
 
-      if (isset($cart->billing) && empty($cart->billing->street)) {
+      if (isset($cart->billing) && empty($cart->billing->customer_address_id) && empty($cart->billing->street)) {
         unset($cart->billing);
       }
 
