diff --git a/modules/acq_sku/acq_sku.drush.inc b/modules/acq_sku/acq_sku.drush.inc
index 28706055b..9df977037 100644
--- a/modules/acq_sku/acq_sku.drush.inc
+++ b/modules/acq_sku/acq_sku.drush.inc
@@ -49,6 +49,14 @@ function acq_sku_drush_command() {
     ],
   ];
 
+  $commands['commerce-cats-remove-duplicates'] = [
+    'description' => 'Remove all duplicate categories available in system.',
+    'aliases'     => ['acccrd'],
+    'examples'    => [
+      'drush acccrd' => 'Remove all duplicate categories available in system.',
+    ],
+  ];
+
   $commands['clean-synced-data'] = [
     'description' => 'Flush all commerce data from the site (Products, SKUs, Product Categories and Product Options).',
     'aliases'     => ['accd'],
@@ -141,6 +149,76 @@ function drush_acq_sku_sync_commerce_product_options() {
 }
 
 /**
+ * Implements drush commerce-cats-remove-duplicates command.
+ */
+function drush_acq_sku_commerce_cats_remove_duplicates() {
+  drush_print(dt('Cleaning all commerce categories, please wait...'));
+
+  $db = \Drupal::database();
+
+  /** @var \Drupal\taxonomy\TermStorageInterface $termStorage */
+  $termStorage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+
+  /** @var \Drupal\node\NodeStorageInterface $nodeStorage */
+  $nodeStorage = \Drupal::entityTypeManager()->getStorage('node');
+
+  $query = $db->select('taxonomy_term__field_commerce_id','ttfci');
+  $query->addField('ttfci', 'field_commerce_id_value', 'commerce_id');
+  $query->groupBy('commerce_id');
+  $query->having('count(*) > 1');
+  $result = $query->execute()->fetchAllKeyed(0, 0);
+
+  if (empty($result)) {
+    drush_print(dt('No duplicate categories found.'));
+    return;
+  }
+
+  foreach ($result as $commerce_id) {
+    drush_print(dt('Duplicate categories found for commerce id: @commerce_id.', [
+      '@commerce_id' => $commerce_id,
+    ]));
+
+    $query = $db->select('taxonomy_term__field_commerce_id','ttfci');
+    $query->addField('ttfci', 'entity_id', 'tid');
+    $query->condition('ttfci.field_commerce_id_value', $commerce_id);
+    $query->orderBy('tid', 'DESC');
+    $tids = $query->execute()->fetchAllKeyed(0, 0);
+
+    foreach ($tids as $tid) {
+      $query = $nodeStorage->getQuery();
+      $query->condition('field_category', $tid);
+      $nodes = $query->execute();
+
+      if (empty($nodes)) {
+        drush_print(dt('No nodes found for tid: @tid for commerce id: @commerce_id. Deleting', [
+          '@commerce_id' => $commerce_id,
+          '@tid' => $tid,
+        ]));
+
+        $term = $termStorage->load($tid);
+        $term->delete();
+
+        unset($tids[$tid]);
+
+        //
+        if (count($tids) == 1) {
+          break;
+        }
+      }
+      else {
+        drush_print(dt('@count nodes found for tid: @tid for commerce id: @commerce_id. Not Deleting', [
+          '@commerce_id' => $commerce_id,
+          '@tid' => $tid,
+          '@count' => count($nodes),
+        ]));
+      }
+    }
+  }
+
+  drush_print(dt('Done.'));
+}
+
+/**
  * Implements drush clean synced data command.
  */
 function drush_acq_sku_clean_synced_data() {
diff --git a/modules/acq_sku/src/ConductorCategoryRepository.php b/modules/acq_sku/src/ConductorCategoryRepository.php
index 9d46ad328..1d5650f8a 100644
--- a/modules/acq_sku/src/ConductorCategoryRepository.php
+++ b/modules/acq_sku/src/ConductorCategoryRepository.php
@@ -93,20 +93,17 @@ public function loadCategoryTerm($commerce_id) {
 
     $tids = $query->execute();
 
-    if (count($tids) > 1) {
-
-      $this->logger->error('Multiple terms found for category id @cid', ['@cid' => $commerce_id]);
-
+    if (count($tids) == 0) {
       return (NULL);
-
     }
-    elseif (count($tids) == 1) {
-      $term = $this->termStorage->load(array_shift($tids));
-      $this->terms[$commerce_id] = $term;
-      return ($term);
+    elseif (count($tids) > 1) {
+      $this->logger->error('Multiple terms found for category id @cid', ['@cid' => $commerce_id]);
     }
 
-    return (NULL);
+    $term = $this->termStorage->load(array_shift($tids));
+    $this->terms[$commerce_id] = $term;
+
+    return ($term);
   }
 
   /**
diff --git a/modules/acq_sku/acq_sku.drush.inc b/modules/acq_sku/acq_sku.drush.inc
index 9df977037..80caaddc0 100644
--- a/modules/acq_sku/acq_sku.drush.inc
+++ b/modules/acq_sku/acq_sku.drush.inc
@@ -200,7 +200,9 @@ function drush_acq_sku_clean_commerce_cats() {
 
         unset($tids[$tid]);
 
-        //
+        // Break the loop if only one left now, we might not have any products
+        // added yet and categories are synced which means there will be no
+        // nodes for any term.
         if (count($tids) == 1) {
           break;
         }
diff --git a/modules/acq_sku/acq_sku.drush.inc b/modules/acq_sku/acq_sku.drush.inc
index 9df977037..fdd61c968 100644
--- a/modules/acq_sku/acq_sku.drush.inc
+++ b/modules/acq_sku/acq_sku.drush.inc
@@ -136,7 +136,6 @@ function drush_acq_sku_sync_commerce_cats() {
   drush_print(dt('Done.'));
 }
 
-
 /**
  * Implements drush product attribute sync command.
  */
@@ -162,7 +161,7 @@ function drush_acq_sku_clean_commerce_cats() {
   /** @var \Drupal\node\NodeStorageInterface $nodeStorage */
   $nodeStorage = \Drupal::entityTypeManager()->getStorage('node');
 
-  $query = $db->select('taxonomy_term__field_commerce_id','ttfci');
+  $query = $db->select('taxonomy_term__field_commerce_id', 'ttfci');
   $query->addField('ttfci', 'field_commerce_id_value', 'commerce_id');
   $query->groupBy('commerce_id');
   $query->having('count(*) > 1');
@@ -178,7 +177,7 @@ function drush_acq_sku_clean_commerce_cats() {
       '@commerce_id' => $commerce_id,
     ]));
 
-    $query = $db->select('taxonomy_term__field_commerce_id','ttfci');
+    $query = $db->select('taxonomy_term__field_commerce_id', 'ttfci');
     $query->addField('ttfci', 'entity_id', 'tid');
     $query->condition('ttfci.field_commerce_id_value', $commerce_id);
     $query->orderBy('tid', 'DESC');
@@ -200,7 +199,9 @@ function drush_acq_sku_clean_commerce_cats() {
 
         unset($tids[$tid]);
 
-        //
+        // Break the loop if only one left now, we might not have any products
+        // added yet and categories are synced which means there will be no
+        // nodes for any term.
         if (count($tids) == 1) {
           break;
         }
diff --git a/modules/acq_sku/src/ConductorCategoryRepository.php b/modules/acq_sku/src/ConductorCategoryRepository.php
index 1d5650f8a..9f7e4e934 100644
--- a/modules/acq_sku/src/ConductorCategoryRepository.php
+++ b/modules/acq_sku/src/ConductorCategoryRepository.php
@@ -3,7 +3,6 @@
 namespace Drupal\acq_sku;
 
 use Drupal\Core\Entity\EntityTypeManagerInterface;
-use Drupal\Core\Entity\Query\QueryFactory;
 use Drupal\Core\Logger\LoggerChannelFactory;
 
 /**
