diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index e5b9f4af..21439974 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -173,6 +173,10 @@ public function addItemToCart($sku, $quantity) {
     $items[] = ['sku' => $sku, 'qty' => $quantity];
 
     $this->cart->items = $items;
+
+    // We changed items in cart, lets ask user to go through the checkout flow
+    // again to ensure everything is updated properly.
+    $this->setCheckoutStep('');
   }
 
   /**
@@ -190,6 +194,10 @@ public function removeItemFromCart($sku) {
       }
     }
     $this->cart->items = $items;
+
+    // We changed items in cart, lets ask user to go through the checkout flow
+    // again to ensure everything is updated properly.
+    $this->setCheckoutStep('');
   }
 
   /**
@@ -225,6 +233,10 @@ public function addItemsToCart(array $items) {
    */
   public function setItemsInCart(array $items) {
     $this->cart->items = $items;
+
+    // We changed items in cart, lets ask user to go through the checkout flow
+    // again to ensure everything is updated properly.
+    $this->setCheckoutStep('');
   }
 
   /**
@@ -245,6 +257,10 @@ public function updateItemQuantity($sku, $quantity) {
     }
 
     $this->cart->items = $items;
+
+    // We changed items in cart, lets ask user to go through the checkout flow
+    // again to ensure everything is updated properly.
+    $this->setCheckoutStep('');
   }
 
   /**
