diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index e5b9f4af..60269966 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -110,6 +110,14 @@ public function customerEmail() {
    */
   public function totals() {
     if (isset($this->cart, $this->cart->totals)) {
+      // Remove the comma coming from Magento, we will add while formatting.
+      // Casting to float or using floatval stops at comma and doesn't return
+      // whole value.
+      // NumberFormatter.parse requires locale, we don't have expected value
+      // available anywhere in the system for now (kw_AR) so not using it.
+      foreach ($this->cart->totals as &$value) {
+        $value = str_replace(',', '', $value);
+      }
       return $this->cart->totals;
     }
     return NULL;
