diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index e5b9f4af..323fb001 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -110,6 +110,10 @@ public function customerEmail() {
    */
   public function totals() {
     if (isset($this->cart, $this->cart->totals)) {
+      // Remove the comma coming from Magento, we will add while formatting.
+      foreach ($this->cart->totals as &$value) {
+        $value = str_replace(',', '', $value);
+      }
       return $this->cart->totals;
     }
     return NULL;
