diff --git a/acq_commerce.module b/acq_commerce.module
index 16662b89..1f4dab81 100644
--- a/acq_commerce.module
+++ b/acq_commerce.module
@@ -51,6 +51,13 @@ function template_preprocess_acq_commerce_price(array &$variables) {
   // Fetch the currency code position.
   $variables['currency_code_position'] = $config->get('currency_code_position');
 
+  // Remove the comma coming from Magento, we will add while formatting.
+  // Casting to float or using floatval stops at comma and doesn't return
+  // whole value.
+  // NumberFormatter.parse requires locale, we don't have expected value
+  // available anywhere in the system for now (kw_AR) so not using it.
+  $variables['price'] = str_replace(',', '', $variables['price']);
+
   $variables['price'] = number_format($variables['price'], $config->get('decimal_points'));
 }
