diff --git a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index bc36a640e..9887625f7 100644
--- a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -206,7 +206,14 @@ public function addToCartSubmit(array &$form, FormStateInterface $form_state) {
           ]
       ));
 
-      $cart->addItemToCart($tree_pointer->getSku(), $quantity);
+      $cart->addRawItemToCart([
+        'name' => $label,
+        'sku' => $tree['parent']->getSKU(),
+        'qty' => $quantity,
+        'options' => [
+          'configurable_item_options' => $options,
+        ],
+      ]);
 
       // Add child SKU to form state to allow other modules to use it.
       $form_state->setTemporaryValue('child_sku', $tree_pointer->getSKU());
