diff --git a/modules/acq_sku/src/ProductOptionsManager.php b/modules/acq_sku/src/ProductOptionsManager.php
index e70a7420b4..f462716b69 100644
--- a/modules/acq_sku/src/ProductOptionsManager.php
+++ b/modules/acq_sku/src/ProductOptionsManager.php
@@ -132,6 +132,11 @@ protected function createProductOption($langcode, $option_id, $option_value, $at
     if ($term = $this->loadProductOptionByOptionId($attribute_code, $option_id, NULL, FALSE)) {
       $save_term = FALSE;
 
+      // Save term even if weight changes.
+      if ($term->getWeight() != $weight) {
+        $save_term = TRUE;
+      }
+
       if ($term->hasTranslation($langcode)) {
         $term = $term->getTranslation($langcode);
 
