diff --git a/modules/acq_cybersource/src/Controller/CybersourceController.php b/modules/acq_cybersource/src/Controller/CybersourceController.php
index a203b3851..4a0ad94eb 100644
--- a/modules/acq_cybersource/src/Controller/CybersourceController.php
+++ b/modules/acq_cybersource/src/Controller/CybersourceController.php
@@ -139,10 +139,10 @@ public function getToken() {
     // Set the payment method.
     $cart->setPaymentMethod('cybersource');
 
-    // Update the cart.
-    $this->cartStorage->updateCart(FALSE);
-
     try {
+      // Update the cart.
+      $this->cartStorage->updateCart(FALSE);
+
       $token_info = $this->apiWrapper->cybersourceTokenRequest($cart_id, $cc_type);
 
       // Do some cleaning.
