diff --git a/modules/acq_sku/src/Entity/SKU.php b/modules/acq_sku/src/Entity/SKU.php
index 095d6e0c5c..c75886aa1c 100644
--- a/modules/acq_sku/src/Entity/SKU.php
+++ b/modules/acq_sku/src/Entity/SKU.php
@@ -11,6 +11,7 @@
 use Drupal\acq_commerce\SKUInterface;
 use Drupal\Core\StringTranslation\TranslatableMarkup;
 use Drupal\file\Entity\File;
+use Drupal\file\FileInterface;
 use Drupal\user\UserInterface;
 
 /**
@@ -671,4 +672,20 @@ public function clearStockCache() {
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function postDelete(EntityStorageInterface $storage, array $entities) {
+    parent::postDelete($storage, $entities);
+
+    // Delete media files.
+    foreach ($entities as $entity) {
+      /** @var \Drupal\acq_sku\Entity\SKU $sku  */
+      foreach ($entity->getMedia() as $media) {
+        if ($media['file'] instanceof FileInterface) {
+          $media['file']->delete();
+        }
+      }
+    }
+  }
 }
