diff --git a/modules/acq_cart/src/Form/CustomerCartForm.php b/modules/acq_cart/src/Form/CustomerCartForm.php
index 878e1a093..d530b5fd2 100644
--- a/modules/acq_cart/src/Form/CustomerCartForm.php
+++ b/modules/acq_cart/src/Form/CustomerCartForm.php
@@ -66,7 +66,12 @@ public function getFormId() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $cart = $this->cartStorage->getCart();
+    $cart = $this->cartStorage->getCart(FALSE);
+
+    if (empty($cart)) {
+      return $form;
+    }
+
     $items = NULL;
 
     if ($cart) {
