diff --git a/modules/acq_cart/src/Form/CustomerCartForm.php b/modules/acq_cart/src/Form/CustomerCartForm.php
index 7b210ff..7e9e00a 100644
--- a/modules/acq_cart/src/Form/CustomerCartForm.php
+++ b/modules/acq_cart/src/Form/CustomerCartForm.php
@@ -117,7 +117,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ];
     }
 
-    if ((float) $totals['discount'] > 0) {
+    if ((float) $totals['discount'] != 0) {
       $form['totals']['discount'] = [
         'label' => ['#plain_text' => t('Discount')],
         'value' => ['#plain_text' => $totals['discount']],
