diff --git a/modules/acq_cybersource/src/Controller/CybersourceController.php b/modules/acq_cybersource/src/Controller/CybersourceController.php
index 4a0ad94eb6..ef14c5d3dd 100644
--- a/modules/acq_cybersource/src/Controller/CybersourceController.php
+++ b/modules/acq_cybersource/src/Controller/CybersourceController.php
@@ -117,16 +117,13 @@ public function getToken() {
       throw new \InvalidArgumentException(sprintf('Invalid credit cart type %s or type not allowed.', $type));
     }
 
-    $cart_id = $this->cartStorage->getCartId(FALSE);
+    // Get the cart object.
+    $cart = $this->cartStorage->getCart(FALSE);
 
-    // We must have a valid cart in session.
-    if (empty($cart_id)) {
+    if (empty($cart)) {
       throw new AccessDeniedHttpException('No cart available to get token');
     }
 
-    // Get the cart object.
-    $cart = $this->cartStorage->getCart(FALSE);
-
     // Allow all modules to validate and update cart data before doing getToken.
     $errors = [];
     \Drupal::moduleHandler()->alter('acq_cybersource_before_get_token_cart', $cart, $form_data, $errors);
@@ -143,7 +140,7 @@ public function getToken() {
       // Update the cart.
       $this->cartStorage->updateCart(FALSE);
 
-      $token_info = $this->apiWrapper->cybersourceTokenRequest($cart_id, $cc_type);
+      $token_info = $this->apiWrapper->cybersourceTokenRequest($cart->id(), $cc_type);
 
       // Do some cleaning.
       foreach ($token_info as &$info) {
@@ -167,7 +164,7 @@ public function getToken() {
     catch (\Exception $e) {
       $this->logger->info('Error while getting Cybersource token: %message <br> Cart id: %cart_id and Card type: %card_type', [
         '%message' => $e->getMessage(),
-        '%cart_id' => $cart_id,
+        '%cart_id' => $cart->id(),
         '%card_type' => $cc_type,
       ]);
 
