diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index a37db7f7ef..281df7311b 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -184,6 +184,10 @@ public function post(array $products = []) {
         $has_bundle = $query->execute();
 
         if (!$has_bundle) {
+          $this->logger->warning('Product type @type not supported.', [
+            '@type' => $product['type'],
+          ]);
+          $ignored++;
           continue;
         }
 
