diff --git a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
index f7ae8418..94f68014 100644
--- a/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
+++ b/modules/acq_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php
@@ -303,6 +303,7 @@ public function deriveProductTree(SKU $sku) {
       $sku->field_configurable_attributes->getString()
     );
 
+    $tree['configurables'] = [];
     foreach ($configurables as $configurable) {
       $tree['configurables'][$configurable['code']] = $configurable;
     }
