diff --git a/modules/acq_sku/acq_sku.api.php b/modules/acq_sku/acq_sku.api.php
new file mode 100644
index 0000000..2be6bcd
--- /dev/null
+++ b/modules/acq_sku/acq_sku.api.php
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @file
+ * Hooks specific to the acq_sku module.
+ */
+
+/**
+ * @addtogroup hooks
+ * @{
+ */
+
+/**
+ * Alter product node before it is saved during insert or update.
+ *
+ * Product data from API is passed here to allow other modules to read from
+ * the data provided by API and update product node accordingly.
+ *
+ * @param \Drupal\node\NodeInterface $node
+ *   Node to alter.
+ * @param array $product
+ *   Array containing details provided by API.
+ */
+function hook_acq_sku_product_node_alter(Drupal\node\NodeInterface $node, array $product) {
+
+}
+
+/**
+ * @} End of "addtogroup hooks".
+ */
diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index 9d2362a..4931fd1 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -288,6 +288,9 @@ private function createDisplayNode(array $product)

     $node->setPublished(FALSE);

+    // Invoke the alter hook to allow all modules to update the node.
+    \Drupal::moduleHandler()->alter('acq_sku_product_node', $node, $product);
+
     return($node);
   }

@@ -358,6 +361,10 @@ private function updateNodeCategories(array $nids, array $product)
       $node = $this->entityManager->getStorage('node')->load($nid);
       if ($node && $node->id()) {
         $node->field_category = $categories;
+
+        // Invoke the alter hook to allow all modules to update the node.
+        \Drupal::moduleHandler()->alter('acq_sku_product_node', $node, $product);
+
         $node->save();
       }
     }
