diff --git a/modules/acq_sku/src/Entity/SKU.php b/modules/acq_sku/src/Entity/SKU.php
index f63a1bc29..e414de95a 100644
--- a/modules/acq_sku/src/Entity/SKU.php
+++ b/modules/acq_sku/src/Entity/SKU.php
@@ -106,28 +106,12 @@ public function getMedia($reset = FALSE) {
     if ($media_data = $this->get('media')->getString()) {
       $update_sku = FALSE;
 
-      $media_data_full = unserialize($media_data);
+      $media_data = unserialize($media_data);
 
-      if (empty($media_data_full)) {
+      if (empty($media_data)) {
         return [];
       }
 
-      // @TODO: Remove this hard coded fix after getting answer why we have empty
-      // second array index and why all media come in first array index.
-      $media_data = reset($media_data_full);
-
-      // Sort media data by position.
-      usort($media_data, function ($a, $b) {
-        $position1 = (int) $a['position'];
-        $position2 = (int) $b['position'];
-
-        if ($position1 == $position2) {
-          return 0;
-        }
-
-        return ($position1 < $position2) ? -1 : 1;
-      });
-
       foreach ($media_data as &$data) {
         if ($data['media_type'] == 'image') {
           if (empty($data['fid'])) {
@@ -153,10 +137,7 @@ public function getMedia($reset = FALSE) {
       }
 
       if ($update_sku) {
-        // @TODO: Remove this hard coded fix after getting answer why we have
-        // empty second array index and why all media come in first array index.
-        $media_data_full[0] = $media_data;
-        $this->get('media')->setValue(serialize($media_data_full));
+        $this->get('media')->setValue(serialize($media_data));
         $this->save();
       }
     }
diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index d729163f6..0db3fb49c 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -205,7 +205,10 @@ public function post(array $products = []) {
       $sku->special_price->value = $product['special_price'];
       $sku->final_price->value = $product['final_price'];
       $sku->attributes = $this->formatProductAttributes($product['attributes']);
-      $sku->media = serialize($product['extension']['media']);
+
+      // Update product media to set proper position.
+      $sku->media = $this->getProcessedMedia($product);
+
       $sku->attribute_set = $product['attribute_set_label'];
       $sku->product_id = $product['product_id'];
 
@@ -351,6 +354,9 @@ private function formatProductAttributes(array $attributes) {
     $formatted = [];
 
     foreach ($attributes as $name => $value) {
+      if (is_array($value)) {
+        continue;
+      }
 
       if (!strlen($value)) {
         continue;
@@ -462,4 +468,41 @@ private function updateFields($parent, SKU $sku, array $values) {
     }
   }
 
+  protected function getProcessedMedia($product) {
+    $media = [];
+
+    if (isset($product['extension'], $product['extension']['media'])) {
+      $media = $product['extension']['media'];
+
+      if (isset($product['attributes'], $product['attributes']['image'])) {
+        $image = $product['attributes']['image'];
+
+        // @TODO: Remove this hard coded fix after getting answer why we have
+        // empty second array index and why all media come in first array index.
+        $media = reset($media);
+
+        foreach ($media as &$data) {
+          if (substr_compare($data['file'], $image, -strlen($image) ) === 0) {
+            $data['position'] = -1;
+            break;
+          }
+        }
+      }
+
+      // Sort media data by position.
+      usort($media, function ($a, $b) {
+        $position1 = (int) $a['position'];
+        $position2 = (int) $b['position'];
+
+        if ($position1 == $position2) {
+          return 0;
+        }
+
+        return ($position1 < $position2) ? -1 : 1;
+      });
+    }
+
+    return serialize($media);
+  }
+
 }
