diff --git a/modules/acq_sku/acq_sku.install b/modules/acq_sku/acq_sku.install
index 433715eb2..89e4e70a4 100644
--- a/modules/acq_sku/acq_sku.install
+++ b/modules/acq_sku/acq_sku.install
@@ -11,6 +11,21 @@
 
 /**
  * Implements hook_update_N().
+ *
+ * Add additional indexes to improve performance.
+ */
+function acq_sku_update_8004() {
+  foreach (_acq_sku_indexes() as $index => $data) {
+    if (!\Drupal::database()->schema()->indexExists($data['table'], $index)) {
+      \Drupal::database()
+        ->schema()
+        ->addIndex($data['table'], $index, $data['fields'], $data['spec']);
+    }
+  }
+}
+
+/**
+ * Implements hook_update_N().
  */
 function acq_sku_update_8003() {
   $config_path = drupal_get_path('module', 'acq_sku') . '/config/install/acq_sku.configurable_form_settings.yml';
@@ -45,28 +60,7 @@ function acq_sku_update_8002() {
  * Add index for SKU and langcode on acq_sku_field_data table.
  */
 function acq_sku_update_8001() {
-  // Add index for SKU and langcode on acq_sku_field_data table.
-  \Drupal::database()->schema()->addIndex('acq_sku_field_data',
-    'acq_sku_sku_langcode',
-    [
-      ['sku', 64],
-      ['langcode', 10],
-    ],
-    [
-      'fields' => [
-        'sku' => [
-          'type' => 'varchar',
-          'length' => 255,
-          'default' => '',
-        ],
-        'langcode' => [
-          'type' => 'varchar',
-          'length' => 255,
-          'default' => '',
-        ],
-      ],
-    ]
-  );
+  // Done in 8004 again.
 }
 
 /**
@@ -93,26 +87,49 @@ function acq_sku_install() {
     'weight' => 0,
   ])->save();
 
+  foreach (_acq_sku_indexes() as $index => $data) {
+    if (!\Drupal::database()->schema()->indexExists($data['table'], $index)) {
+      \Drupal::database()
+        ->schema()
+        ->addIndex($data['table'], $index, $data['fields'], $data['spec']);
+    }
+  }
+}
+
+/**
+ * Function to get indexes to be added for SKU related tables.
+ *
+ * @return array
+ *   Array containing indexes to be added for SKU related tables.
+ */
+function _acq_sku_indexes() {
+  $indexes = [];
+
   // Add index for first 30 character of SKU on acq_sku_field_data table.
-  \Drupal::database()->schema()->addIndex('acq_sku_field_data', 'acq_sku_field_sku', [['sku', 64]], [
+  $indexes['acq_sku_field_sku'] = [
+    'table' => 'acq_sku_field_data',
     'fields' => [
-      'sku' => [
-        'type' => 'varchar',
-        'length' => 255,
-        'default' => '',
+      ['sku', 64],
+    ],
+    'spec' => [
+      'fields' => [
+        'sku' => [
+          'type' => 'varchar',
+          'length' => 255,
+          'default' => '',
+        ],
       ],
     ],
-  ]);
-
+  ];
 
   // Add index for SKU and langcode on acq_sku_field_data table.
-  \Drupal::database()->schema()->addIndex('acq_sku_field_data',
-    'acq_sku_sku_langcode',
-    [
+  $indexes['acq_sku_sku_langcode'] = [
+    'table' => 'acq_sku_field_data',
+    'fields' => [
       ['sku', 64],
       ['langcode', 10],
     ],
-    [
+    'spec' => [
       'fields' => [
         'sku' => [
           'type' => 'varchar',
@@ -125,6 +142,54 @@ function acq_sku_install() {
           'default' => '',
         ],
       ],
-    ]
-  );
+    ],
+  ];
+
+  $indexes['acq_sku__attr_join'] = [
+    'table' => 'acq_sku__attributes',
+    'fields' => [
+      'entity_id',
+      ['attributes_key', 128],
+    ],
+    'spec' => [
+      'fields' => [
+        'entity_id' => [
+          'type' => 'integer',
+        ],
+        'attributes_key' => [
+          'type' => 'varchar',
+          'length' => 255,
+          'default' => '',
+        ],
+      ],
+    ],
+  ];
+
+  $indexes['acq_sku__attr_where'] = [
+    'table' => 'acq_sku__attributes',
+    'fields' => [
+      'entity_id',
+      ['attributes_key', 128],
+      ['attributes_value', 128],
+    ],
+    'spec' => [
+      'fields' => [
+        'entity_id' => [
+          'type' => 'integer',
+        ],
+        'attributes_key' => [
+          'type' => 'varchar',
+          'length' => 255,
+          'default' => '',
+        ],
+        'attributes_value' => [
+          'type' => 'varchar',
+          'length' => 255,
+          'default' => '',
+        ],
+      ],
+    ],
+  ];
+
+  return $indexes;
 }
