diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 380912c..ac37acb 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -532,6 +532,29 @@ public function getPaymentToken($method) {
   }
 
   /**
+   * Function to subscribe an email for newsletter.
+   *
+   * @param string $email
+   *   E-Mail to subscribe.
+   */
+  public function subscribeNewsletter($email) {
+    $endpoint = 'newsletter/subscribe';
+
+    $doReq = function ($client, $opt) use ($endpoint, $email) {
+      $opt['form_params']['email'] = $email;
+
+      return ($client->post($endpoint, $opt));
+    };
+
+    try {
+      return $this->tryAgentRequest($doReq, 'subscribeNewsletter');
+    }
+    catch (ConductorException $e) {
+      throw new \Exception($e->getMessage(), $e->getCode());
+    }
+  }
+
+  /**
    * Preforms a test call to conductor.
    *
    * @return array
