diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index c520bccb9..d729163f6 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -141,6 +141,11 @@ public function post(array $products = []) {
     foreach ($products as $product) {
       $langcode = acq_commerce_get_langcode_from_store_id($product['store_id']);
 
+      // Magento might have stores that what we don't support.
+      if (empty($langcode)) {
+        continue;
+      }
+
       if ($debug && !empty($debug_dir)) {
         // Export product data into file.
         if (!isset($fps) || !isset($fps[$langcode])) {
