diff --git a/src/Conductor/IngestAPIWrapper.php b/src/Conductor/IngestAPIWrapper.php
index 7310132..008e1c2 100644
--- a/src/Conductor/IngestAPIWrapper.php
+++ b/src/Conductor/IngestAPIWrapper.php
@@ -33,16 +33,22 @@ public function __construct(ClientFactory $client_factory, ConfigFactoryInterfac
    * Performs full conductor sync.
    */
   public function productFullSync() {
-    $endpoint = $this->apiVersion . "/ingest/product/sync";
-
-    $doReq = function ($client, $opt) use ($endpoint) {
-      return $client->post($endpoint, $opt);
-    };
-
-    try {
-      $this->tryIngestRequest($doReq, 'productFullSync', 'products');
-    }
-    catch (ConductorException $e) {
+    foreach (acq_commerce_get_store_language_mapping() as $langcode => $store_id) {
+      if (empty($store_id)) {
+        continue;
+      }
+
+      $endpoint = $this->apiVersion . '/ingest/product/sync?store_id=' . $store_id;
+
+      $doReq = function ($client, $opt) use ($endpoint) {
+        return $client->post($endpoint, $opt);
+      };
+
+      try {
+        $this->tryIngestRequest($doReq, 'productFullSync', 'products');
+      }
+      catch (ConductorException $e) {
+      }
     }
   }
 
