diff --git a/modules/acq_sku/src/Entity/SKU.php b/modules/acq_sku/src/Entity/SKU.php
index abdc13958..8f59d89c3 100644
--- a/modules/acq_sku/src/Entity/SKU.php
+++ b/modules/acq_sku/src/Entity/SKU.php
@@ -246,16 +246,29 @@ public function getPluginInstance() {
    *   Found SKU
    */
   public static function loadFromSku($sku, $langcode = '', $log_not_found = TRUE, $create_translation = FALSE) {
-    if (empty($langcode)) {
+    $is_multilingual = \Drupal::languageManager()->isMultilingual();
+
+    if ($is_multilingual && empty($langcode)) {
       $langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
     }
 
     $storage = \Drupal::entityTypeManager()->getStorage('acq_sku');
     $skus = $storage->loadByProperties(['sku' => $sku]);
 
-    if (count($skus) > 1) {
-      \Drupal::logger('acq_sku')->error('Duplicate SKUs found while loading for @sku.', ['@sku' => $sku]);
+    // First check if we have some result before doing anything else.
+    if (count($skus) == 0) {
+      // We don't log the error while doing sync.
+      if ($log_not_found) {
+        \Drupal::logger('acq_sku')->error('No SKU found for @sku.', ['@sku' => $sku]);
+      }
+
+      return NULL;
+    }
 
+    // Remove all skus in other languages if there are more than one available
+    // We need to have multiple languages enabled to clean db result.
+    // If only one is available, we might be adding translation, leave it as is.
+    if ($is_multilingual && !empty($skus) && count($skus) > 1) {
       // Get rid of undesired languages. Later first one is picked up.
       foreach ($skus as $key => $sku) {
         if ($sku->langcode->value != $langcode) {
@@ -264,18 +277,14 @@ public static function loadFromSku($sku, $langcode = '', $log_not_found = TRUE,
       }
     }
 
-    if (count($skus) == 0) {
-      // We don't log the error while doing sync.
-      if ($log_not_found) {
-        \Drupal::logger('acq_sku')->error('No SKU found for @sku.', ['@sku' => $sku]);
-      }
-
-      return NULL;
+    // Log error message if we have more than one available even after cleanup.
+    if (count($skus) > 1) {
+      \Drupal::logger('acq_sku')->error('Duplicate SKUs found while loading for @sku.', ['@sku' => $sku]);
     }
 
     $sku_entity = array_shift($skus);
 
-    if (\Drupal::languageManager()->isMultilingual()) {
+    if ($is_multilingual) {
       if ($sku_entity->hasTranslation($langcode)) {
         $sku_entity = $sku_entity->getTranslation($langcode);
       }
