diff --git a/src/Conductor/APIWrapper.php b/src/Conductor/APIWrapper.php
index 5e37081..63e97d3 100644
--- a/src/Conductor/APIWrapper.php
+++ b/src/Conductor/APIWrapper.php
@@ -2,6 +2,8 @@
 
 namespace Drupal\acq_commerce\Conductor;
 
+use Drupal\Core\Logger\RfcLogLevel;
+
 /**
  * APIWrapper class.
  */
@@ -85,7 +87,8 @@ public function skuStockCheck($sku) {
       }
     }
     catch (ConductorException $e) {
-      throw new \Exception($e->getMessage(), $e->getCode());
+      // Log the stock error, do not throw error if stock info is missing.
+      \Drupal::logger('acq_commerce')->log(RfcLogLevel::ERROR, 'Unable to get the stock for @sku : @message', ['@sku' => $sku, '@message' => $e->getMessage()]);
     }
 
     return $stock;
