diff --git a/modules/acq_cart/src/Cart.php b/modules/acq_cart/src/Cart.php
index 0ca499f..9a48d2d 100644
--- a/modules/acq_cart/src/Cart.php
+++ b/modules/acq_cart/src/Cart.php
@@ -326,10 +326,11 @@ public function setShipping($address) {
   /**
    * {@inheritdoc}
    */
-  public function setShippingMethod($carrier, $method) {
+  public function setShippingMethod($carrier, $method, array $extension = []) {
     $this->cart->carrier = [
       'carrier_code' => $carrier,
       'method_code' => $method,
+      'extension' => $extension,
     ];
   }
 
diff --git a/modules/acq_cart/src/CartInterface.php b/modules/acq_cart/src/CartInterface.php
index efe1472..ceb3f56 100644
--- a/modules/acq_cart/src/CartInterface.php
+++ b/modules/acq_cart/src/CartInterface.php
@@ -159,8 +159,10 @@ public function getShippingMethodAsString();
    *   The current shipping carrier.
    * @param string $method
    *   The current shipping method.
+   * @param array $extension
+   *   Additional data passed in extension.
    */
-  public function setShippingMethod($carrier, $method);
+  public function setShippingMethod($carrier, $method, array $extension = []);
 
   /**
    * Sets the shipping address.
