diff --git a/modules/acq_sku/acq_sku.module b/modules/acq_sku/acq_sku.module
index 648a11018..66defbcca 100644
--- a/modules/acq_sku/acq_sku.module
+++ b/modules/acq_sku/acq_sku.module
@@ -113,7 +113,9 @@ function acq_sku_theme_suggestions_acq_sku(array $variables) {
  *   Cache id.
  */
 function acq_sku_get_stock_cache_id(SKU $sku_entity) {
-  return 'stock:' . $sku_entity->language()->getId() . ':' . $sku_entity->id();
+  // We use only SKU for cache id, it will always be the same for same SKU
+  // in any store (language in Drupal).
+  return 'stock:' . $sku_entity->getSku();
 }
 
 /**
