diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index 82aaa2c05..6afef5912 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -135,6 +135,8 @@ public function post(array $products = []) {
     $created = 0;
     $updated = 0;
     $failed = 0;
+    $ignored = 0;
+    $deleted = 0;
 
     $config = $this->configFactory->get('acq_commerce.conductor');
     $debug = $config->get('debug');
@@ -212,7 +214,7 @@ public function post(array $products = []) {
           // Delete the SKU.
           $sku->delete();
 
-          $updated++;
+          $deleted++;
           continue;
         }
 
@@ -222,7 +224,7 @@ public function post(array $products = []) {
       else {
         if ($product['status'] != 1) {
           $this->logger->info('Not creating disabled SKU in system: @sku.', ['@sku' => $product['sku']]);
-          $failed++;
+          $ignored++;
           continue;
         }
 
@@ -338,10 +340,12 @@ public function post(array $products = []) {
     }
 
     $response = [
-      'success' => (bool) (($created > 0) || ($updated > 0)),
+      'success' => !$failed && ($created || $updated || $ignored || $deleted),
       'created' => $created,
       'updated' => $updated,
       'failed' => $failed,
+      'ignored' => $ignored,
+      'deleted' => $deleted,
     ];
 
     return (new ResourceResponse($response));
diff --git a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
index 6afef5912..9cdbe7948 100644
--- a/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
+++ b/modules/acq_sku/src/Plugin/rest/resource/ProductSyncResource.php
@@ -176,14 +176,14 @@ public function post(array $products = []) {
 
       if (!isset($product['sku']) || !strlen($product['sku'])) {
         $this->logger->error('Invalid or empty product SKU.');
-        $failed++;
+        $ignored++;
         continue;
       }
 
       // Don't import configurable SKU if it has no configurable options.
       if ($product['type'] == 'configurable' && empty($product['extension']['configurable_product_options'])) {
         $this->logger->error('Empty configurable options for SKU: @sku', ['@sku' => $product['sku']]);
-        $failed++;
+        $ignored++;
         continue;
       }
