diff --git a/src/Conductor/ConductorResultException.php b/src/Conductor/ConductorResultException.php
index 5cba2d5cf..4463f3955 100644
--- a/src/Conductor/ConductorResultException.php
+++ b/src/Conductor/ConductorResultException.php
@@ -47,6 +47,12 @@ public function __construct(array $result) {
         $response = json_decode($responseString, TRUE);
         if (is_array($response) && isset($response['message'])) {
           $mesg = $response['message'];
+
+          if (isset($response['parameters'])) {
+            foreach ($response['parameters'] as $name => $value) {
+              $mesg = str_replace("%$name", $value, $mesg);
+            }
+          }
         }
       }
 
