diff --git a/src/SearchApiAlgoliaHelper.php b/src/SearchApiAlgoliaHelper.php
index ec51ce2..580528f 100644
--- a/src/SearchApiAlgoliaHelper.php
+++ b/src/SearchApiAlgoliaHelper.php
@@ -81,12 +81,20 @@ class SearchApiAlgoliaHelper {
     // Remove the search items for all the entity's translations.
     foreach ($indexes as $index) {
       $object_id_field = $index->getOption('object_id_field');
+      $apply_suffix = $index->getOption('algolia_index_apply_suffix');
       if ($object_id_field) {
         $object_id = $entity->get($object_id_field)->getString();
 
         if ($object_id) {
           // @todo make this work with indexes having language suffix.
-          $this->scheduleForDeletion($index, [$object_id]);
+          if($apply_suffix){
+            foreach ($this->languageManager->getLanguages() as $language) {
+              $this->scheduleForDeletion($index, [$object_id], $language->getId());
+            }
+          }
+          else {
+            $this->scheduleForDeletion($index, [$object_id]);
+          }
         }
       }
     }
@@ -112,7 +120,7 @@ class SearchApiAlgoliaHelper {
       $indexId = $index->getOption('algolia_index_name');
       if ($index->getOption('algolia_index_apply_suffix') && !empty($language)) {
         $objectIdParsed = explode(':', $objectid);
-        if (end($objectIdParsed) !== $language) {
+        if (end($objectIdParsed) !== $language && empty($index->getOption('object_id_field'))) {
           // Skip the object ids in other language.
           continue;
         }
